---
title: "Metabolites reflect variability introduced by mesoscale eddies in the North Pacific Subtropical Gyre"
author:
  - William Kumler,$^1$ Wei Qin,$^{1+}$ Rachel A. Lundeen,$^1$ Benedetto Barone,$^2$ Laura T. Carlson,$^1$ Anitra E. Ingalls$^{1*}$
output: 
  word_document:
    reference_docx: Frontiers_Template.docx
bibliography:
- Exported Items.bib
- r-packages.bib
csl: frontiers-in-marine-science.csl
---

$^1$School of Oceanography, University of Washington, Seattle, WA, USA

$^2$Daniel K. Inouye Center for Microbial Oceanography: Research and Education and Department of Oceanography, University of Hawai'i at Mānoa, Honolulu, HI, USA

$^*$Correspondence: Anitra E. Ingalls, aingalls\@uw.edu

$^+$Present Address: School of Biological Sciences, Institute for Environmental Genomics, University of Oklahoma, Norman, OK, USA

Word counts:

  - Main text (excludes the abstract, section titles, figure and table captions, funding statement, acknowledgments, and references): [X,XXX]
  - Number of figures: [X]
  - Number of tables: [X]

Running head: Metabolites reflect variability introduced by eddies

```{r Run analysis, include=FALSE}
if(!file.exists("../mzMLs/neg/180205_Poo_TruePoo_Full1.mzML")){
  init_timeout <- options("timeout")$timeout
  options(timeout=3600) # Allow 1 hour for each FTP download

  download.file("ftp://www.metabolomicsworkbench.org/Studies/ST002788_HILIC_NEG.zip",
                destfile = "../mzMLs/ST002788_HILIC_NEG.zip")
  unzip("../mzMLs/ST002788_HILIC_NEG.zip", exdir = "../mzMLs/neg")
  file.remove("../mzMLs/ST002788_HILIC_NEG.zip")
  download.file("ftp://www.metabolomicsworkbench.org/Studies/ST002789_HILIC_NEG.zip",
                destfile = "../mzMLs/ST002789_HILIC_NEG.zip")
  unzip("../mzMLs/ST002789_HILIC_NEG.zip", exdir = "../mzMLs/neg")
  file.remove("../mzMLs/ST002789_HILIC_NEG.zip")
  download.file("ftp://www.metabolomicsworkbench.org/Studies/ST002788_HILIC_POS.zip",
                destfile = "../mzMLs/ST002788_HILIC_POS.zip")
  unzip("../mzMLs/ST002788_HILIC_POS.zip", exdir = "../mzMLs/pos")
  file.remove("../mzMLs/ST002788_HILIC_POS.zip")
  download.file("ftp://www.metabolomicsworkbench.org/Studies/ST002789_HILIC_POS.zip",
                destfile = "../mzMLs/ST002789_HILIC_POS.zip")
  unzip("../mzMLs/ST002789_HILIC_POS.zip", exdir = "../mzMLs/pos", unzip="unzip")
  file.remove("../mzMLs/ST002789_HILIC_POS.zip")
  options(timeout=init_timeout)
}

if(!file.exists("../metadata/filled_file_metadata.csv")){
  setwd("../metadata")
  knitr::purl("metadata_control.Rmd")
  source("metadata_control.R")
  file.remove("metadata_control.R")
  setwd("../manuscript")
}

if(!file.exists("../targeted/all_peaks.csv")){
  setwd("../targeted")
  knitr::purl("../targeted/targeted_control.Rmd")
  source("targeted_control.R")
  file.remove("targeted_control.R")
  setwd("../manuscript")
}

if(!file.exists("../untargeted/all_peaks.csv")){
  setwd("../untargeted")
  knitr::purl("untargeted_control.Rmd")
  data_iters <- c("FK_pos", "FK_neg", "MS_pos", "MS_neg")
  for(cruise in c("FK", "MS")){
    for(polarity in c("pos", "neg")){
      source("untargeted_control.R")
    }
  }
  file.remove("../untargeted/untargeted_control.R")
  map(data_iters, function(output_dir){
    comb_peaks <- read_csv(paste0(output_dir, "_output/clean_peak_data.csv")) %>%
      mutate(temp=output_dir) %>%
      separate(temp, into = c("cruise", "polarity"), sep = "_") %>%
      mutate(feature=str_replace(feature, "FT", paste0(toupper(output_dir), "_")))
  }) %>%
    bind_rows() %>%
    write_csv("all_peaks.csv")
  map(data_iters, function(output_dir){
    comb_peaks <- read_csv(paste0(output_dir, "_output/clean_feat_data.csv")) %>%
      mutate(temp=output_dir) %>%
      separate(temp, into = c("cruise", "polarity"), sep = "_") %>%
      mutate(feature=str_replace(feature, "FT", paste0(toupper(output_dir), "_")))
  }) %>%
    bind_rows() %>%
    write_csv("all_feats.csv")
  setwd("../manuscript")
}
```

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE, message = FALSE, warning = FALSE, eval = FALSE)
library(tidyverse)
library(RaMS)
library(vegan)
library(cowplot)
library(viridis)
library(scales)
library(ggnewscale)
library(egg)
library(broom)
library(lme4)
library(openxlsx)
library(readxl)
library(ggh4x)
library(ggtext)
library(gridExtra)
library(openxlsx)
library(arrow)

if(!dir.exists("supplement"))dir.create("supplement")

filled_file_metadata <- read_csv("../metadata/filled_file_metadata.csv") %>%
  filter(polarity=="pos") %>%
  mutate(local_time=as.POSIXct(time, tz="HST")) %>%
  select(-polarity)
clean_stans <- read_csv("../metadata/clean_stans.csv") %>%
  mutate(n_C=as.numeric(str_extract(formula, "(?<=C)\\d+"))) %>%
  mutate(n_N=as.numeric(str_extract(formula, "(?<=N)\\d+"))) %>%
  mutate(n_N=case_when(
    !is.na(n_N)~n_N,
    str_detect(formula, "N")~1,
    TRUE~0
  ))

untarg_feats <- read_csv("../untargeted/all_feats.csv")
untarg_peaks <- read_csv("../untargeted/all_peaks.csv") %>%
  filter(str_detect(filename, "_Smp_"))
targ_peaks <- read_csv("../targeted/all_peaks.csv") %>%
    mutate(filename=paste0(filename, ".mzML"))

metadata_levels <- c(
    "no23_um", "po4_um", "si_um", "PC_um", "PN_um", "DON_um", "DOP_um", "oxy_um",
    "chl_ug", "phaeo_ug", "alk", "dic", "ph", "theta",
    paste0(c("pro", "syn", "euk", "het"), "_abund"), "beam_atten",
    "abs_depth", "time", "lat", "sla", "local_hour", 
    "TDN", "TDP", "temp", "pressure", "sal", "sigma",
    "ctd_chl"
  )
metadata_labels <- c(
    "Nitrate + nitrite", "Phosphate", "Silicate", "Particulate C", "Particulate N",
    "Dissolved organic N", "Dissolved organic P", "Oxygen", "Chlorophyll a",
    "Phaeophytin", "Alkalinity", "Dissolved inorganic C", "pH", "Density",
    paste(c("Pro.", "Syn.", "Picoeuk.", "Het. bact."), "abundance"),
    "Beam attenuation",
    "Depth", "Time", "Latitude", "Sea level anomaly", "Local hour", 
    "Total dissolved nitrogen", "Total dissolved phosphorus",
    "Temperature", "Pressure", "Salinity", "Sigma-theta",
    "Chlorophyll (fluor.)"
  )
metadata_twoline_labels <- c(
    "Nitrate + nitrite", "Phosphate", "Silicate", "Particulate C", "Particulate N",
    "Dissolved\norganic N", "Dissolved\norganic P", "Oxygen", "Chlorophyll a",
    "Phaeophytin", "Alkalinity", "Dissolved\ninorganic C", "pH", "Density",
    paste(c("Prochlorococcus", "Synechococcus", "Picoeukaryote", 
            "Heterotrophic bact."), "\nabundance"),
    "Beam attenuation",
    "Depth", "Time", "Latitude", "Sea level anomaly", "Local hour",
    "Total dissolved\nnitrogen", "Total dissolved\nphosphorus",
    "Temperature", "Pressure", "Salinity", "Sigma-theta", "Chlorophyll\n(fluorescence)"
  )
metadata_short_labels <- c("N+N", "PO4", "Si", "PC", "PN", "DON", "DOP", "O2", "Chl a",
                  "Phaeophytin", "Alk", "DIC", "pH", "Density", 
                  paste(c("Pro", "Syn", "Picoeuk", "Het. bact."), "abund"),
                  "Beam atten.", "Depth", "Time", "Latitude", "SLA", "Local hour",
                  "Tot. N", "Tot. P", "Temp", "dB", "Sal", "σ-θ", "CTD chl.")

splividis <- function(n, ...){
  vir <- viridis::viridis(n, ...)
  if(n%%2==0){
    vir[as.numeric(matrix(1:n, ncol=2, byrow = TRUE))]
  } else {
    index_vec <- suppressWarnings(as.numeric(matrix(1:n, ncol=2, byrow = TRUE)))
    vir[head(index_vec, -1)]
  }
}
```

# Abstract

Mesoscale eddies significantly alter open ocean environments such as those found in the subtropical gyres that cover a large fraction of the global ocean. Previous studies have explored eddy effects on biogeochemistry and microbial community composition but not on the molecular composition of particulate organic matter. This study reports the absolute concentration of 67 metabolites and relative abundances for 640 molecular features to understand how mesoscale eddies impact the metabolome of the North Pacific Subtropical Gyre during two cruises in 2017 and 2018. We find that many metabolites track biomass trends, but metabolites like isethionic acid, homarine, and trigonelline linked to eukaryotic phytoplankton were enriched at the deep chlorophyll maximum of the cyclonic features, while degradation products such as arsenobetaine were enriched in anticyclones. In every analysis, metabolites with the strongest responses were detected via untargeted mass spectrometry, indicating that the molecules most sensitive to environmental perturbation were not among the characterized metabolome. By analyzing depth variability (accounting for 20-40% of metabolomic variability across ~150 meters) and the vertical displacement of isopycnal surfaces (explaining 10-20% of variability across a sea level anomaly range of 40 centimeters and a spatial distance of 300 kilometers), this analysis constrains the importance of mesoscale eddies in shaping the chemical composition of particulate matter in the largest biomes on the planet.


# Introduction

High frequency observations at Station ALOHA in the North Pacific Subtropical Gyre (NPSG) over the past 25 years have revealed temporal and spatial variability in what had previously been considered a relatively homogenous environment [@Karl1999; @Karl2017; @Karl2021]. A major source of variability comes in the form of mesoscale eddies in which water is entrained into circular surface currents tens to hundreds of kilometers in diameter [@Karl2017; @McGillicuddy2016]. 

Eddies can be observed via satellite altimetry, which measures anomalies in sea surface height. Those that have a positive sea level anomaly (SLA) typically indicate regions where deep water layers and isopycnal surfaces are depressed and are associated with clockwise rotation in the Northern Hemisphere. In contrast, a negative SLA corresponds to counterclockwise rotation in the Northern Hemisphere and the uplift of deep water layers into the sunlit region of the ocean [@McGillicuddy2016]. Mode-water eddies are an exception to the conventions established above [@Sweeney2003; @McGillicuddy2007] but here we only focus on the cyclonic (negative SLA) and anticyclonic (positive SLA) mesoscale features that are commonly observed near station ALOHA [@Barone2019].

The uplift of deep, nutrient-rich seawater into the euphotic zone alters microbial communities [@Rii2022]. Measurements of chlorophyll in cyclones reveal a shallower and more intense maximum [@Barone2022; @Cornec2021] as a result of eukaryotic phytoplankton thriving in the higher nutrient concentrations while cyanobacterial biomass is reduced [@Hawco2021]. This growth of large eukaryotes corresponds to a net increase in biomass and primary productivity [@Benitez-Nelson2007], especially near the deep chlorophyll maximum [@Barone2022]. In contrast, anticyclones produce conditions favorable to nitrogen fixation by accumulating diazotrophs such as *Crocosphaera* and *Trichodesmium* [@Fong2008; @Olson2015; @Dugenne2020; @Dugenne2023].

This shift in community structure and productivity should result in a corresponding shift in the composition of the particulate organic matter produced. Eukaryotic organisms have distinct chemical fingerprints from those of cyanobacteria and heterotrophic bacteria [@Heal2021; @Durham2022] and the relative contribution of these different taxa to total biomass should reflect their chemical composition. However, recent work from @Harke2021 showed that overall community function may be robust across eddy types at the surface while @Gleich2024 detected significant differences in protistan community composition and metabolic potential down to 250 meters, with heterotrophy-associated protistan transcripts enriched in the cyclone. The chemical composition of organic matter is one metric of community function since community metabolism and interactions are in part mediated through organic molecules, making their measurement useful for determining shifts in community dynamics.

In this work, we directly tested for differences in the chemical composition of particulate matter due to changes in eddy state in the NPSG. We sampled from cyclonic and anticyclonic eddies spatially near each other and used targeted and untargeted mass spectrometry to explore changes in the composition of small, polar molecules. We expected to find that overall metabolite abundance would reflect shifts in biomass over multiple depths. We also expected to see enrichment in the deep chlorophyll maximum of cyclonic features for those metabolites especially abundant in eukaryotic organisms. Finally, we predicted that these patterns would be robust across years and sampling regimes for a reliable way to link these ocean features with the chemical composition of organic matter in the open ocean.

# Materials and Methods

## Cruise information
Samples were collected from two cruises in the North Pacific Subtropical Gyre near Station ALOHA that targeted strong mesoscale eddy features as described in @Dugenne2023 and @Gleich2024 (Figure 1). Briefly, the 2017 MESO-SCOPE cruise (Microbial Ecology of the Surface Ocean-Simons Collaboration on Ocean Processes and Ecology, KM1709 on the R/V *Kilo Moana*) consisted of a transect across adjacent cyclonic and anticyclonic eddies as well as two long-term Lagrangian stations at the center of each eddy. The cyclonic eddy had a maximum negative sea level anomaly (SLA) of -20 centimeters and the anticyclonic eddy reached +24 centimeters. The transect samples were taken at various times of day as the ship transected the adjacent eddies while the eddy center samples were all collected between 5 and 8 pm. In 2018, the Hawaiian Eddy Experiment (HEE, FK180310 on the R/V *Falkor*) targeted new cyclonic and anticyclonic eddies in approximately the same location with samples taken at the center of each eddy (maximum negative anomaly in the cyclone = -15 cm, maximum positive anomaly in the anticyclone = +26 cm) (Figure 1).

![](figures/fig_1_MapForWill_v3.jpg)

*Figure 1: Sampling during MESO-SCOPE and the Hawaiian Eddy Experiment (HEE). Cruise bounds are shown in the large central map with Station ALOHA colored as a point in red near the Hawaiian Islands. Yellow stars denote sampling locations and station numbers relative to the sea level anomaly contours in the background during 28 June 2017 for the MESO-SCOPE cruise (left) and the 6 April 2018 for the HEE cruise (right). Lagrangian sampling near the eddy centers of the MESO-SCOPE cruise was performed by following drifters deployed in proximity of Stations 6 and 12.*

## Biogeochemical data

Biogeochemical measurements were collected as described in @Barone2022 and @Dugenne2023 mostly following protocols used by the HOT program (http://hahana.soest.hawaii.edu/hot/methods/results.html). Briefly, all environmental data was collected via CTD rosette except SLA which was measured via satellite. Particulate carbon and particulate nitrogen were measured using an elemental analyzer. Nitrate + nitrite (N+N) was measured on an autoanalyzer except where concentrations were below 100 nM in which case they were measured using chemiluminescence. Soluble reactive phosphorus (phosphate, PO$_4^{3-}$) was measured on an autoanalyzer or following the magnesium induced coprecipitation method. Values were interpolated linearly to the depths at which metabolite samples were taken using all data collected at the given station. Two particulate carbon values were determined to be spurious, with concentrations 2-3 times higher than typical Station ALOHA values, and were instead estimated using a best-fit linear regression against beam attenuation. CTD data is available here in Supplemental Tables 1 and 2.

```{r supplemental CTD data}
wb <- createWorkbook(creator = "William Kumler, wkumler@uw.edu")
addWorksheet(wb, sheetName = "Table 1 CTD Cast list")
ctd_cast_list <- read_csv("../metadata/complete_ctddata.csv") %>%
  mutate(cruise=ifelse(time<"2018-01-01", "MESOSCOPE", "HEE")) %>%
  select(cruise, everything())
writeData(x=ctd_cast_list, wb = wb, sheet = "Table 1 CTD Cast list")

addWorksheet(wb, sheetName = "Table 2 CTD Cast data by bottle")
ctd_cast_data <- read_csv("../metadata/all_core_gof.csv") %>%
  mutate(cruise=ifelse(cruise=="FK", "HEE", "MESOSCOPE")) %>%
  select(cruise, everything())
writeData(x=ctd_cast_data, wb = wb, sheet = "Table 2 CTD Cast data by bottle")

addWorksheet(wb, sheetName = "Table 3 Interpolated CTD data")
writeData(x=filled_file_metadata, wb = wb, sheet = "Table 3 Interpolated CTD data")
saveWorkbook(wb, file = "supplement/Supplemental Tables.xlsx", overwrite = TRUE)
```

## Sample collection for particulate metabolites
Samples were obtained using the onboard CTD rosette to collect water from 15 meters, the deep chlorophyll maximum (DCM), and 175 meters during the MESO-SCOPE eddy transect and from the DCM ±10 and ±20 meters during the eddy center sampling in the MESO-SCOPE cruise and from 25 meters and the DCM during the HEE cruise. The DCM was determined visually from fluorometer data during the CTD downcast and Niskin bottles were tripped during the return trip to the surface. Seawater from each depth was sampled in triplicate by firing one Niskin bottle for each sample. Samples were brought to the surface and decanted into prewashed (3x with DI, 3x with sampled seawater) polycarbonate bottles for filtration. 10L samples were filtered by peristaltic pump onto 142mm 0.2 µm Durapore filters held by polycarbonate filter holders on a Masterflex tubing line. Pressures were kept as low as possible while still producing a reasonable rate of flow through the filter, approximately 250-500 mL per minute. Samples were then removed from the filter holder using solvent-washed tweezers and placed into pre-combusted aluminum foil packets that were then flash-frozen in liquid nitrogen before being stored at -80 °C until extraction. A methodological blank was also collected by running filtrate through a new filter and then treated identically to the samples.

## Metabolite sample extraction
Extraction followed a modified Bligh & Dyer approach as detailed in @Boysen2018. Briefly, filters were randomized and added to PTFE centrifuge tubes with a 1:1 mix of 100 µm and 400 µm silica beads, approximately 2 mL -20 °C Optima-grade dichloromethane, and approximately 3 mL -20 °C 1:1 methanol/water solution (both also Optima-grade). Isotope labeled extraction standards were also added during this step (Supplemental Table 4). The samples were then bead-beaten three times, followed by triplicate extraction of the aqueous layer into a separate vial after spontaneous separation and replacement with fresh methanol/water mixture using a glass Pasteur pipette with additional bead-beatings in between. The aqueous fraction was then dried down under ultrapure nitrogen gas and warmed using a Fisher-Scientific Reacti-Therm module. Dry samples were reconstituted in 380 µL Optima-grade water and amended with 20 µL isotope-labeled injection standards (Supplemental Table 4) to measure the variability introduced by chromatography and ionization. The reconstituted fraction was syringe-filtered into precombusted glass inserts in HPLC vials to remove any potential clogging material. Samples were then additionally diluted 1:1 with Optima-grade water to prevent overloading on the column and to reduce salt effects. 

A pooled sample was created by combining 20 µL of each sample into the same HPLC vial, and a 1:1 dilution with water created a half-strength pooled sample from that to assess matrix effects and obscuring variation [@Boysen2018]. Also run alongside the environmental samples were samples of authentic standards split into two mixes (Supplemental Table 5). Standards were dissolved in Optima-grade water to help with identification and into an aliquot of the pooled sample to quantify matrix response factors. HPLC vials containing the samples were frozen at -80 °C until thawing shortly before injection.

## HPLC-MS methods
Separate HPLC-MS runs were used for the MESO-SCOPE eddy transect, the MESO-SCOPE eddy center, and the HEE cruise data. Eddy center samples were run in February 2018, eddy transect samples were run later during August of that year, and the HEE samples were run in July 2019. Each run was treated as a single batch and injected in sequence while maintaining the randomization that had occurred during extraction to minimize chromatographic shifts from solvent or column switching.

For each run, a Waters Acquity I-Class UPLC with a SeQuant ZIC-pHILIC column (5 µm particle size, 2.1 mm x 150 mm, from Millipore) was used with 10 mM ammonium carbonate in 85:15 acetonitrile to water (Solvent A) and 10 mM ammonium carbonate in 85:15 water to acetonitrile (Solvent B) at a flow rate of 0.15 mL/min. The column was held at 100% A for 2 minutes, ramped to 64% B over 18 minutes, ramped to 100% B over 1 minute, held at 100% B for 5 minutes, and equilibrated at 100% A for 25 minutes (50 minutes total). The column was maintained at 30 °C. The injection volume was 2 µL for samples and standard mixes. When starting a batch, the column was equilibrated at the starting conditions for at least 30 minutes. To improve the performance of the HILIC column, we maintained the same injection volume, kept the instrument running water blanks between samples as necessary, and injected standards in a representative matrix (the pooled sample) in addition to standards in water. After each batch, the column was flushed with 10 mM ammonium carbonate in 85:15 water to acetonitrile for 20 to 30 minutes.

The Waters Acquity UPLC was coupled to a Thermo Q Exactive HF hybrid Orbitrap high resolution mass spectrometer equipped with a heated electrospray ionization source (H-ESI). The H-ESI voltage was set to 3.3 kV and sheath gas, auxiliary gas, and sweep gas flow rates were set at 16, 3, and 1, respectively. The capillary and auxiliary gas heater temperatures were maintained at 320°C and 100°C, respectively. Full scan analyses were performed with polarity switching and a scan range of 60 to 900 *m/z* at a resolution of 60,000. The instrument was mass calibrated at 200 *m/z* before each run began and once again during the eddy transect sample set to ensure calibrations were always performed within 3-4 days. All data files were then converted to an open-source mzML format and vendor centroided via Proteowizard’s msConvert tool (version 3.0.19297, @Chambers2012). All mzML files have been uploaded to Metabolomics Workbench under Project ID PR001738.

## Targeted metabolomic methods

Compounds for which we had an authentic standard (Supplemental Table 5) were manually integrated using Skyline (versions 4.1 and 21.2.0.470, @Adams2020) and MSDIAL (version 4.36, @Tsugawa2015) with the standard mixes used to ensure the correct peak was integrated by matching retention time. Compounds were removed from the analysis if they were visually assessed to have poor peak quality in the samples or peak areas in the methodological blanks similar to those in the samples. Skyline was used for both the eddy center samples and the HEE cruise data while MSDIAL was used for the eddy transect data due to the larger number of files in that dataset. Raw peak areas were then normalized to their best-matched internal standard (Supplemental Table 4) per @Boysen2018 except that all compounds were normalized to their best match no matter how minimal the improvement. While significant analytical drift was detected during the longer eddy transect sample run, the use of internal standards was able to negate the instrument's general increase in sensitivity. Normalized peaks were then calibrated via a single-point standard curve as determined uniquely for each compound from the authentic standard mixes and used to convert normalized peak area into moles per µL injected and scaled to the amount of seawater filtered to provide a final estimate of environmental concentration for each compound in moles per liter of seawater filtered. Scripts used for this analysis are available at https://github.com/wkumler/MesoscopeMetabolomicsManuscript/tree/master/untargeted.

```{r supp table 1 standards list}
clean_stans_supp <- read_csv("../metadata/clean_stans.csv")
raafay_notes <- read_csv("../targeted/raw_data/Raafay_HILIC_notes.csv") %>%
  select(compound_name, in_samps)

IS_table <- clean_stans_supp %>%
  filter(compound_type=="Internal Standard") %>%
  select(compound_name, formula, stan_mz, stan_rt, conc_um, date_added, polarity) %>%
  mutate(ms_run=case_when(
    date_added<180205~"All",
    date_added<=180821~"MESO-SCOPE Eddy Transect and HEE",
    date_added<=190715~"HEE only",
    TRUE~"None"
  )) %>%
  filter(ms_run!="None") %>%
  mutate(polarity=ifelse(polarity=="pos", "Positive", "Negative")) %>% 
  mutate(stan_type=case_when(
    compound_name%in%c("L-Cysteic acid, 2H3", "3-Sulfolactate, 13C3", "Sulfoacetic acid, 13C2", "Isethionic acid, 13C2", "Taurine, 2H4")~"Extraction",
    TRUE~"Internal"
  )) %>%
  arrange(stan_mz) %>%
  select(`Compound name`=compound_name, `Molecular formula`=formula,
         `m/z ratio`=stan_mz, `Retention time (min)`=stan_rt,
         `Concentration (uM)`=conc_um, `Polarity integrated`=polarity,
         `Which runs`=ms_run, `Extraction or internal`=stan_type)

Auth_table <- clean_stans_supp %>%
  mutate(ms_run=case_when(
    date_added<180205~"All",
    date_added<=180821~"MESO-SCOPE Eddy Transect and HEE",
    date_added<=190715~"HEE only",
    TRUE~"None"
  )) %>%
  filter(ms_run!="None") %>%
  left_join(raafay_notes) %>%
  mutate(polarity=ifelse(polarity=="pos", "Positive", "Negative")) %>%
  arrange(stan_mz) %>%
  select(`Compound name`=compound_name, `Molecular formula`=formula,
         `m/z ratio`=stan_mz, `Retention time (min)`=stan_rt,
         `Concentration (uM)`=conc_um, `Polarity integrated`=polarity,
         `Standard mix`=mix, `Which runs`=ms_run, `Good quality`=in_samps)

wb <- loadWorkbook("supplement/Supplemental Tables.xlsx")
addWorksheet(wb, sheetName = "Table 4 Internal Standards")
writeData(x=IS_table, wb = wb, sheet = "Table 4 Internal Standards")

addWorksheet(wb, sheetName = "Table 5 Authentic Standards")
writeData(x=Auth_table, wb = wb, sheet = "Table 5 Authentic Standards")

saveWorkbook(wb, file = "supplement/Supplemental Tables.xlsx", overwrite = TRUE)
```

## Untargeted metabolomic methods

Samples were also processed via the untargeted workflow detailed in @Kumler2023. In short, XCMS (version 3.10.0, @R-xcms) was used for feature detection with CentWave peakpicking, Obiwarp alignment, and peak density correspondence/grouping. This was followed by manual inspection of the extracted ion chromatogram for each mass feature by an expert for quality with only those peaks classified as "Good" used in the analysis here. Features were then matched with and normalized to an internal standard (Supplemental Table 4) as described above. Scripts used for this analysis are available at https://github.com/wkumler/MesoscopeMetabolomicsManuscript/tree/master/untargeted.

## Statistics

All statistics were run in R version 4.4.0. We used multivariate statistics provided by the vegan package (version 2.6-6.1, @R-vegan) to assess the overall impact of various metrics across the metabolome followed by univariate statistics to investigate individually significant responses. Non-parametric tests were used when data were unlikely to obey parametric assumptions and permutational statistics were preferred wherever possible. Multiple comparison problems were controlled where necessary using a false-discovery rate correction [@Benjamini1995]. For the analyses involving the eddy transect samples, sea level anomaly was treated as a continuous variable with the exception of the PERMANOVA, for which SLA was categorized as cyclonic if less than -5 cm and anticyclonic if greater than 5 cm, with a third category for stations in between [@Anderson2001]. Both the MESO-SCOPE eddy center and HEE analyses treated SLA as a categorical variable. Where time of day was included as an explanatory factor in the PERMANOVA, it was binned into 6-hour increments starting from 3AM (e.g. Midday contains casts collected between 9AM and 3PM). Marginal effects are reported for the PERMANOVA analyses to avoid influencing the results with the order of the terms in the model formula. Total sample size (*n*) for the eddy transect was 33 per depth (3 biological triplicates at each of 11 stations), while the eddy centers had an *n* of 15 per eddy (3 biological replicates at 5 depths) and the HEE data had an *n* of 24 (3 biological replicates at 4 stations (2 in the cyclone and 2 in the anticyclone) at 2 depths each).

# Results

We explored the impacts of isopycnal uplift and depression on the composition of particulate organic matter across pairs of mesoscale eddies of opposite polarity in the North Pacific Subtropical Gyre (NPSG). The first eddy pair was the focus of the July 2017 cruise, Microbial Ecology of the Surface Ocean-Simons Collaboration on Ocean Processes and Ecology (MESO-SCOPE, KM1709 aboard the R/V *Kilo Moana*). During the first phase of our two-phase expedition, a transect was taken from the north to the south consisting of 11 stations across a pair of eddies (the same sampling sites as in @Barone2022 and @Dugenne2023, Figure 1). Samples for particulate metabolomic analysis were collected from 15 meters, the deep chlorophyll maximum (DCM), and 175 meters at each station onto 0.2 µm filters. The next phase of the MESO-SCOPE cruise focused on Lagrangian sampling in the two eddy centers, starting with the cyclone (SLA = -14 cm) before progressing to the anticyclone (SLA = 24 cm). There, metabolomics samples were taken from the DCM as well as 10 and 20 meters above and below it. The second eddy pair was targeted during the March 2018 Hawaiian Eddy Experiment (HEE, FK180310 aboard the R/V *Falkor*). During this followup cruise, a strong anticyclone (SLA = 21 cm) and a nearby cyclone (SLA = -13 cm) were targeted for a similar set of biogeochemical measurements described in @Dugenne2023 and @Gleich2024 (Figure 1). Metabolomics samples were taken from the center of each eddy as described above at both 25 meters and the DCM. We analyzed these three different datasets separately and discuss the findings from each in sequence below.

## Metabolome variability across adjacent mesoscale eddies of opposing polarity during MESO-SCOPE explained by sea level anomaly variations

The metabolome clearly differed between cyclonic and anticyclonic samples, though the magnitude of this difference varied by depth. The largest differences in particulate matter composition were detected in the DCM and 175 meter samples, as shown by the NMDS in Figure 2. 

At the DCM, approximately 14% of the variation in the metabolome could be explained by sea level anomaly (SLA) alone (PERMANOVA, p = 0.002). Additionally, SLA was strongly correlated with the first principal component (PC) of the metabolome  with a Pearson's *r* of 0.615 and 27% of the variance explained by PC1 (Supplemental Figure 1A), indicating that this was one of the largest sources of variation in the dataset. Notably, the samples taken from the exact center of the cyclonic eddy at Station 12 were highly distinct and likely drove much of the explained variance. In the 175 meter samples, 13% of the variance was explained by SLA (PERMANOVA, p = 0.004) and SLA was highly correlated with the second PC of the dataset (*r* = 0.697, % variance explained by PC2 = 15.4, Supplemental Figure 1A), though the first PC did not seem to have any visible pattern with metadata and appeared to largely capture variation between biological triplicates (Supplemental Figure 1B). At 15 meters, SLA trends are slightly less evident with a larger p-value (PERMANOVA, p=0.029) and a lower R$^2$ (0.10) (Figure 2). Time of day was also a significant factor in the PERMANOVA at the surface and DCM (p-values of 0.018 and 0.033, respectively) while the diel effects in the 175 meter samples were unsurprisingly much lower (p = 0.140). The significance of SLA as an explanatory factor was similar even when excluding time of day from the model expression (R$^2_{SLA}$ and p-values within permutation error).

```{r multivar_analysis object creation - nmds/pca/permanova}
set.seed(123)
multivar_analysis <- untarg_peaks %>%
  left_join(filled_file_metadata %>% distinct(ms_run, station, filename, depth), 
          by="filename") %>%
  mutate(depth=factor(depth, levels=c("Surface", "DCM", "Deep"), 
                      labels=c("Surface  (15 m)", "DCM  (110-120 m)", "Deep  (175 m)"))) %>%
  filter(ms_run=="MT") %>%
  arrange(depth) %>%
  nest(data=-depth) %>%
  mutate(wide_peaks=map(data, function(metab_df){
    metab_df %>%
      select(feature, filename, norm_area) %>%
      group_by(feature) %>%
      filter(sum(norm_area==0)<n()*0.9) %>%
      mutate(norm_area=scale(norm_area)[,1]) %>%
      pivot_wider(names_from = "feature", values_from = norm_area) %>%
      column_to_rownames("filename") %>%
      data.matrix()
  })) %>%
  mutate(nmds_output=map(wide_peaks, function(metab_mat){
    env_vec <- pull(filled_file_metadata, sla_corr, filename)[rownames(metab_mat)]
    vegan::metaMDS(metab_mat, k = 2, distance = "manhattan", trace = 0, 
                   autotransform = FALSE, noshare = FALSE, wascores = FALSE) %>%
    vegan::MDSrotate(env_vec)
  })) %>%
  mutate(nmds_stress=map_dbl(nmds_output, pluck, "stress")) %>%
  mutate(perm_output=map(wide_peaks, function(metab_mat){
    adon_expl <- data.frame(filename=rownames(metab_mat)) %>%
      left_join(filled_file_metadata %>% select(filename, sla_class, time_class),
                by="filename")
    vegan::adonis2(
      # metab_mat~sla_class,
      metab_mat~sla_class+time_class,
      data = adon_expl, by = "margin", 
      permutations = 999, method = "manhattan")
  })) %>%
  mutate(perm_r2=map_dbl(perm_output, pluck, "R2", 1)) %>%
  mutate(perm_pval=map_dbl(perm_output, pluck, "Pr(>F)", 1)) %>%
  mutate(pca_output=map(wide_peaks, prcomp)) %>%
  mutate(nmds_stress=paste("s =", round(nmds_stress, digits = 3))) %>%
  mutate(perm_r2=round(perm_r2, digits = 2)) %>%
  mutate(perm_pval=round(perm_pval, digits = 4)) %>%
  mutate(perm_label=paste0("R^2^ = ", perm_r2, "<br>p = ", perm_pval))
```

```{r figure nmds and median metabolite across eddy transect}
nmds_sla_gp <- multivar_analysis %>%
  mutate(nmds_points=map(nmds_output, function(nmds_output_i){
    nmds_output_i %>% pluck("points") %>% as.data.frame()
  })) %>%
  select(depth, nmds_points) %>%
  unnest_longer(nmds_points, indices_to = "filename") %>%
  unnest_wider(nmds_points) %>%
  left_join(filled_file_metadata %>% distinct(filename, sla_corr)) %>%
  mutate(MDS2=ifelse(str_detect(depth,"Deep"), MDS2*-1, MDS2)) %>%
  ggplot() +
  geom_vline(xintercept = 0) +
  geom_hline(yintercept = 0) +
  geom_point(aes(x=MDS1, y=MDS2, fill=sla_corr), color="black", size=4, pch=21) +
  geom_richtext(x=-Inf, y=Inf, aes(label=perm_label), hjust=0, vjust=1, fill="#FFFFFF33",
             data = multivar_analysis, label.r = unit(0, "in"), lineheight = .7) +
  geom_richtext(x=-Inf, y=-Inf, aes(label=nmds_stress), hjust=0, vjust=0, fill="#FFFFFF33",
             data = multivar_analysis, label.r = unit(0, "in")) +
  facet_wrap(~depth, nrow=1) +
  scale_fill_gradient2(
    low = scales::muted("blue"), high = scales::muted("red"),  mid = "grey90",
    aesthetics = c("color", "fill"),
  ) +
  labs(fill="Corr.\nSLA\n(cm)\n", x="NMDS axis 1", y="NMDS axis 2") +
  theme_bw() +
  theme(strip.text = element_text(size = 12)) +
  coord_fixed()

med_metab_df <- untarg_peaks %>%
  left_join(filled_file_metadata %>% distinct(filename, ms_run, station, depth, sla_corr),
            by="filename") %>%
  filter(ms_run=="MT") %>%
  group_by(depth, feature) %>%
  filter(sum(norm_area==0)<n()*0.9) %>%
  mutate(norm_area=scale(norm_area)[,1]) %>%
  ungroup() %>%
  group_by(filename, depth, station, sla_corr) %>%
  summarize(avg_amt=mean(norm_area)) %>%
  ungroup()
# Calculations for correlation between grand median metabolite and SLA
# med_metab_df %>%
#   group_by(station, depth, sla_corr) %>%
#   summarise(grand_avg_amt=mean(avg_amt)) %>%
#   group_by(depth) %>%
#   summarise(sim=cor(sla_corr, grand_avg_amt))
mean_med_metab_df <- med_metab_df %>%
  mutate(station=factor(station, levels=as.character(4:14))) %>%
  mutate(depth=factor(depth, levels=c("Surface", "DCM", "Deep"))) %>%
  group_by(depth, station, sla_corr) %>%
  summarise(mean_val=mean(avg_amt))
med_metab_sla_gp <- med_metab_df %>%
  mutate(station=factor(station, levels=as.character(4:14))) %>%
  mutate(depth=factor(depth, levels=c("Surface", "DCM", "Deep"))) %>%
  ggplot() +
  geom_hline(yintercept = 0) +
  # geom_col(aes(x=station, y=mean_val, fill=sla_corr), color="black", 
  #            data=mean_med_metab_df) +
  geom_point(aes(x=station, y=avg_amt), color="black", size=2) +
  geom_point(aes(x=station, y=mean_val, fill=sla_corr), color="black",
             size=4, pch=21, data=mean_med_metab_df, alpha=0.7) +
  facet_wrap(~depth, nrow=1) +
  scale_color_gradient2(low = scales::muted("blue"), high = scales::muted("red"),
                        mid="grey90", aesthetics = c("color", "fill"), guide = NULL) +
  theme_bw() +
  theme(strip.background = element_blank(), strip.text = element_blank()) +
  labs(x="Station", y="Median metabolite z-score")




# med_metab_sla_gp <- med_metab_df %>%
#   mutate(station=factor(station, levels=as.character(4:14))) %>%
#   mutate(depth=factor(depth, levels=c("Surface", "DCM", "Deep"))) %>%
#   ggplot() +
#   geom_hline(yintercept = 0) +
#   geom_point(aes(x=station, y=avg_amt, fill=sla_corr), color="black", size=4, pch=21) +
#   facet_wrap(~depth, nrow=1) +
#   scale_color_gradient2(low = scales::muted("blue"), high = scales::muted("red"),
#                         mid="grey90", aesthetics = c("color", "fill"), guide = NULL) +
#   theme_bw() +
#   theme(strip.background = element_blank(), strip.text = element_blank()) +
#   labs(x="Station", y="Median metabolite z-score")

library(egg)
nolegend_plot <- ggarrange(nmds_sla_gp+theme(legend.position = "none"), 
                           med_metab_sla_gp, ncol = 1, heights = c(0.7, 0.3), 
                           draw = FALSE)
library(cowplot)
nmds_and_med_metab <- plot_grid(nolegend_plot, get_legend(nmds_sla_gp), 
                                nrow=1, rel_widths = c(0.9, 0.1))
ggsave("fig_2_nmds_and_med_metab.tif", plot = nmds_and_med_metab, device = "tiff", 
       path = "figures", width = 6.875, height = 4.5, units = "in", dpi = 300,
       bg="white")
ggsave("fig_2_nmds_and_med_metab.png", plot = nmds_and_med_metab, device = "png", 
       path = "figures/lowres", width = 6.875, height = 4.5, units = "in", dpi = 144,
       bg="white")
```

![](figures/fig_2_nmds_and_med_metab.tif)

*Figure 2: Distribution of metabolites in multivariate space across adjacent eddies of opposite polarity during the MESO-SCOPE transect, broken down by depth. Top panels depict non-metric multidimensional scaling (NMDS) plots with individual samples colored based on their corrected sea level anomaly. NMDS stress values (s) have been reported in the bottom left corner, while PERMANOVA R$^2$ and p-values are reported in the top left. SLA trends are visible in the DCM and 175 meter samples, with dark blue circles consistently discriminating from the dark red circles along the first multidimensional axis. Bottom panels depict the direction and magnitude of this effect by plotting the mean value of all z-scored metabolites across three biological triplicates in color with the raw values in black behind.*

```{r confirm that analytical order does not affect conclusions, eval=FALSE}
read_csv("../untargeted/IS_integrations/all_is.csv") %>%
  select(filename=Replicate.Name, compound_name=Precursor.Ion.Name, area=Area) %>%
  filter(str_detect(filename, "_Smp_")) %>%
  mutate(filename=paste0(filename, ".mzML")) %>%
  pivot_wider(names_from=compound_name, values_from = "area") %>%
  mutate("Inj_vol"=ifelse(str_detect(filename, "Poo.*Half"), 0.5, 1)) %>%
  pivot_longer(-filename, names_to = "IS_name", values_to = "IS_area") %>%
  filter(IS_name!="Guanosine monophosphate, 15N5") %>%
  left_join(filled_file_metadata %>% select(filename, timestamp, sla_corr, ms_run)) %>%
  filter(ms_run=="MT") %>%
  arrange(timestamp) %>%
  drop_na() %>%
  mutate(filename=fct_inorder(filename)) %>%
  ggplot(aes(x=timestamp, y=IS_area, color=sla_corr)) +
  geom_hline(yintercept = 0) +
  geom_point() +
  facet_wrap(~IS_name, scales="free_y") +
  scale_fill_gradient2(
    low = scales::muted("blue"), high = scales::muted("red"),  mid = "grey90",
    aesthetics = c("color", "fill"),
  ) +
  theme_bw()



read_csv("../untargeted/MS_pos_output/deaddiso_peaks.csv") %>%
  filter(str_detect(filename, "_Smp_")) %>%
  left_join(filled_file_metadata %>% distinct(filename, ms_run, station, depth, sla_corr),
            by="filename") %>%
  filter(ms_run=="MT") %>%
  group_by(depth, feature) %>%
  filter(sum(into==0)<n()*0.9) %>%
  mutate(into=scale(into)[,1]) %>%
  ungroup() %>%
  group_by(filename, depth, station, sla_corr) %>%
  summarize(avg_amt=mean(into)) %>%
  ungroup() %>%
  left_join(filled_file_metadata %>% select(filename, timestamp)) %>%
  arrange(timestamp) %>%
  mutate(filename=fct_inorder(filename)) %>%
  ggplot(aes(x=timestamp, y=avg_amt, color=sla_corr)) +
  geom_hline(yintercept = 0) +
  geom_point() +
  geom_smooth(method="loess", formula=y~x, color="black") +
  scale_fill_gradient2(
    low = scales::muted("blue"), high = scales::muted("red"),  mid = "grey90",
    aesthetics = c("color", "fill"),
  ) +
  theme_bw()


med_metab_by_timestamp <- read_csv("../untargeted/all_peaks.csv") %>%
  filter(str_detect(filename, "_Smp_")) %>%
  left_join(filled_file_metadata %>% distinct(filename, ms_run, station, depth, sla_corr),
            by="filename") %>%
  filter(ms_run=="MT") %>%
  group_by(depth, feature) %>%
  filter(sum(norm_area==0)<n()*0.9) %>%
  mutate(norm_area=scale(norm_area)[,1]) %>%
  ungroup() %>%
  group_by(filename, depth, station, sla_corr) %>%
  summarize(avg_amt=mean(norm_area)) %>%
  ungroup() %>%
  left_join(filled_file_metadata %>% select(filename, timestamp)) %>%
  arrange(timestamp) %>%
  mutate(filename=fct_inorder(filename))

med_metab_by_timestamp %>%
  ggplot(aes(x=timestamp, y=avg_amt, color=sla_corr)) +
  geom_hline(yintercept = 0) +
  geom_point() +
  geom_smooth(method="loess", formula=y~x, color="black") +
  scale_fill_gradient2(
    low = scales::muted("blue"), high = scales::muted("red"),  mid = "grey90",
    aesthetics = c("color", "fill"),
  ) +
  theme_bw()

med_metab_by_timestamp %>%
  mutate(station=factor(station, levels=as.character(4:14))) %>%
  mutate(depth=factor(depth, levels=c("Surface", "DCM", "Deep"))) %>%
  ggplot() +
  geom_hline(yintercept = 0) +
  geom_point(aes(x=station, y=avg_amt, color=timestamp), size=2) +
  facet_wrap(~depth, nrow=1) +
  theme_bw() +
  theme(strip.background = element_blank(), strip.text = element_blank()) +
  labs(x="Station", y="Median metabolite z-score")
```


To characterize the difference between eddies in a lower-dimensional space, we plotted an average metabolite peak area per station at each depth. We performed z-score normalization on each mass feature to give them all equal weight while preserving the variance between samples then calculated the median z-score for each sample, resulting in large positive values when metabolites are more abundant in a given sample than in the median sample (Figure 2). The average metabolite collected at the exact center of the cyclonic eddy (Station 12) had higher values than other samples at this same depth, indicating that most metabolites were more abundant at the cyclone's DCM than elsewhere in the transect. The 175 meter samples, on the other hand, showed the largest metabolite abundances in the anticyclone while the cyclone had consistently lower median z-scores. This method highlighted the negative correlation between the average metabolite and SLA at the DCM ($r_{DCM}= -0.716$) while the 15 meter and 175 meter samples showed the opposite trend ($r_{15m} = 0.211$, $r_{175m} = 0.830$).

```{r perc increase at cyclone DCM center, eval=FALSE}
filled_file_metadata %>%
  filter(ms_run=="MT") %>%
  filter(depth=="DCM") %>%
  distinct(cruise, station, cast, depth, PC_um) %>%
  group_by(MT_DCM_cyc_center=ifelse(station==12, "St12", "Other")) %>%
  summarise(mean_val=mean(PC_um)) %>%
  pivot_wider(names_from=MT_DCM_cyc_center, values_from = mean_val) %>%
  mutate(perc_increase=St12/Other)

targ_peaks %>%
  left_join(filled_file_metadata, by="filename") %>%
  filter(ms_run=="MT") %>%
  filter(samp_type=="Smp") %>%
  filter(depth=="DCM") %>%
  group_by(filename, station, tripl, PC_um) %>%
  summarize(tot_nM=sum(nM)) %>%
  group_by(MT_DCM_cyc_center=ifelse(station==12, "St12", "Other")) %>%
  # group_by(MT_DCM_cyc_center=ifelse(station==12, "St12", "Other"), tripl) %>%
  summarise(mean_val=mean(tot_nM)) %>%
  pivot_wider(names_from=MT_DCM_cyc_center, values_from = mean_val) %>%
  mutate(perc_increase=St12/Other)
```

```{r frac_pc_in_metab_df creation}
frac_pc_in_metab_df <- targ_peaks %>%
  filter(str_detect(filename, "180821_Smp")) %>%
  left_join(clean_stans %>% select(compound_name, formula, polarity, n_C, n_N)) %>%
  mutate(nM_C=n_C*nM, nM_N=n_N*nM) %>%
  group_by(filename) %>%
  summarize(total_nM_C=sum(nM_C), total_nM_N=sum(nM_N)) %>%
  left_join(filled_file_metadata, by = join_by(filename)) %>%
  mutate(frac_C_in_metab=total_nM_C/1000/PC_um,
         frac_N_in_metab=total_nM_N/1000/PN_um) %>%
  mutate(depth=str_extract(depth, "DCM|Deep|Surface")) %>%
  mutate(depth=factor(depth, levels=c("Surface", "DCM", "Deep"))) %>%
  mutate(station=factor(station, levels=c(as.character(4:14))))
```

```{r PC_vs_totnM regression values, include=FALSE}
no_outlier_frac_pc_in_metab_df <- frac_pc_in_metab_df %>%
  filter(!(station=="11" & tripl=="A" & depth=="Surface")) %>%
  ungroup()

# Overall trend between PC and total nM metabolites (all depths)
broom::tidy(lm(total_nM_C~PC_um, data=no_outlier_frac_pc_in_metab_df))[2,]
with(no_outlier_frac_pc_in_metab_df, cor(PC_um, total_nM_C))
lme4::lmer(total_nM_C~PC_um + (1|depth), data=no_outlier_frac_pc_in_metab_df)

# By-depth trend between PC and total nM metabolites
no_outlier_frac_pc_in_metab_df %>%
  group_by(depth) %>%
  summarise(overall=broom::tidy(lm(total_nM_C~PC_um))[2,], 
            cor_vals=cor(PC_um, total_nM_C))


# Same for PN
broom::tidy(lm(total_nM_N~PN_um, data=no_outlier_frac_pc_in_metab_df))[2,]
with(no_outlier_frac_pc_in_metab_df, cor(PN_um, total_nM_N))
lme4::lmer(total_nM_N~PN_um + (1|depth), data=no_outlier_frac_pc_in_metab_df)
no_outlier_frac_pc_in_metab_df %>%
  group_by(depth) %>%
  summarise(overall=broom::tidy(lm(total_nM_N~PN_um))[2,], 
            cor_vals=cor(PN_um, total_nM_N))

filled_file_metadata %>%
  distinct(PC_um, PN_um) %>%
  drop_na() %>%
  cor()
```

This shift in metabolite abundance was largely driven by a corresponding shift in biomass. Particulate carbon and summed metabolite concentration were tightly correlated across all samples and depths (Type I linear regression $\beta = 24.9\pm1.82$ (SE), Pearson's $r$ = 0.813, p-value < 0.001, Supplemental Figure 2). This overall trend was largely driven by higher values at 15 meters and the DCM than at depth, but when analyzed at each depth individually the general correspondence held (Supplemental Figure 2). The DCM and 175 meter samples had the stronger trends ($\beta_{DCM} = 22.6\pm5.53$, $r= 0.593$, p-value < 0.001; $\beta_{175m} = 23.9\pm7.36$, $r= 0.503$, p-value = 0.003) while 15 meter samples showed no significant relationship ($\beta_{15m} = 3.4\pm12.9$, $r= 0.049$, p-value = 0.792). This meant that metabolites contributed a relatively fixed fraction of carbon to the particulate pool with our 53 quantified metabolites representing between 1.5% and 5% of the carbon in the system (Figure 3). Particulate nitrogen was tightly correlated with particulate carbon ($r=0.933$) but the fraction of particulate nitrogen represented by the quantified metabolites was slightly higher across the transect with contributions typically between 2% and 6%. 

Although the untargeted pool contains more molecular features, we expect that a majority of the signal has been captured by our 53 targeted compounds. Of the total peak area that passed quality control, approximately 67% was captured by these 53 molecules. An additional 15% of the total peak area was putatively annotated as various inorganic ions for which no authentic standard was available but could be matched by mass and isotope pattern. Additionally, our targeted list accounted for 13 out of the top 20 largest molecular features by total peak area with an additional 4 features annotated as the inorganic ions for 17/20 of the largest features known. However, these calculations should be taken as estimates because peak area does not correspond directly to environmental concentration and it is entirely possible for abundant environmental compounds to have small peak areas if they ionize poorly on the mass spectrometer or are removed during the sampling and extraction process.

```{r untarg targ comparison, eval=FALSE}
untarg_peaks %>%
  select(feature, filename, norm_area) %>%
  left_join(untarg_feats %>% select(feature, compound_name)) %>%
  group_by(is.na(compound_name)) %>%
  summarise(s=sum(norm_area))

summed_labels <- untarg_peaks %>%
  filter(str_detect(feature, "MS")) %>%
  group_by(feature) %>%
  summarise(s=sum(norm_area)) %>%
  arrange(desc(s)) %>%
  left_join(untarg_feats %>% select(feature, mzmed, rtmed, compound_name, polarity)) %>%
  mutate(compound_name=ifelse(feature=="MS_NEG_0155", "Bromine79", compound_name)) %>%
  mutate(compound_name=ifelse(feature=="MS_NEG_0778", "Iodine", compound_name)) %>%
  mutate(compound_name=ifelse(feature=="MS_NEG_0161", "Sulfite", compound_name)) %>%
  mutate(compound_name=ifelse(feature=="MS_NEG_0039", "Nitrate", compound_name)) %>%
  mutate(compound_name=ifelse(feature=="MS_POS_1122", "TMAO [2M+H]", compound_name))

summed_labels %>%
  group_by(is.na(compound_name)) %>%
  count()
summed_labels %>%
  group_by(is.na(compound_name)) %>%
  summarize(s=sum(s)/1e9)

summed_labels %>%
  filter(feature!="MS_NEG_0055") %>%
  filter(feature!="MS_NEG_0068") %>%
  head(30) %>%
  print(n=Inf)

msdata <- grabMSdata(list.files("../tmzMLs/neg/", full.names=TRUE, pattern="180821_Poo"))
v <- msdata$MS1[mz%between%pmppm(146.0451, 5)]
qplotMS1data(v)
```

Samples taken from the DCM of the cyclone center (Station 12, Figure 2) showed especially high particulate carbon (67% increase, rising from 1.71 μM across all other DCM samples to 2.85 μM at Station 12) and the highest measured total metabolite concentrations (73% higher, increasing from 10.0 nM to 17.2 nM). This explains the large separation observed between these samples and the rest of the data in the NMDS plot of Figure 2.

```{r figure stacked barplot MT eddy transect w PC and SLA}
arrow_df <- data.frame(
  cyc_type=c("Anticyclone\ncenter", "Cyclone\ncenter"),
  x=c("6", "12"), xend=c("6", "12"),
  y=c(5, 4), yend=c(9.5, 6.5),
  color=c(scales::muted("red"), scales::muted("blue")),
  depth=factor("Deep", levels=c("Surface", "DCM", "Deep"), 
               labels=c("Surface  (15 m)", "DCM  (110-120 m)", "Deep  (175 m)")),
  lab_x=c("5", "10")
)
avg_targ_concs <- targ_peaks %>%
  left_join(filled_file_metadata, by="filename") %>%
  filter(ms_run=="MT") %>%
  filter(samp_type=="Smp") %>%
  group_by(compound_name) %>%
  mutate(mean_cmpd_nm=mean(nM)) %>%
  ungroup()
barplot_targ_gp <- avg_targ_concs %>%
  group_by(compound_name, untripl) %>%
  summarise(nM=median(nM), mean_cmpd_nm=unique(mean_cmpd_nm)) %>%
  ungroup() %>%
  arrange(desc(mean_cmpd_nm)) %>%
  mutate(compound_name=ifelse(compound_name=="Trimethylamine N-oxide", "TMAO", compound_name)) %>%
  mutate(compound_name=ifelse(compound_name=="Hydroxyisoleucine", "HO-Ile", compound_name)) %>%
  mutate(compound_name=ifelse(compound_name=="Glycine betaine", "GBT", compound_name)) %>%
  mutate(compound_name=str_remove(compound_name, "^L-")) %>%
  mutate(compound_name=ifelse(
    compound_name%in%head(unique(compound_name), 9), 
    compound_name, "Other")) %>%
  mutate(compound_name=fct_inorder(compound_name)) %>%
  left_join(filled_file_metadata %>% distinct(untripl, depth, station)) %>%
  group_by(depth, station, compound_name) %>%
  summarise(nM=sum(nM), .groups = "keep") %>%
  summarise(mean_nM=mean(nM)) %>%
  mutate(depth=factor(depth, levels=c("Surface", "DCM", "Deep"), 
                      labels=c("Surface  (15 m)", "DCM  (110-120 m)", "Deep  (175 m)"))) %>%
  mutate(station=factor(station, levels=as.character(4:14))) %>%
  ggplot() +
  geom_hline(yintercept = 0) +
  geom_col(aes(x=station, y=mean_nM, fill=compound_name), width = 0.9) +
  scale_fill_manual(values = c(splividis(9), "grey50")) +
  scale_y_continuous(expand = c(0.005, 0.1)) +
  labs(x=NULL, y="Median metabolite\n concentration (nM)", fill=NULL) +
  facet_grid(~depth, scales="free", space = "free") +
  guides(fill=guide_legend(nrow=2, byrow = TRUE)) +
  ggnewscale::new_scale_fill() +
  geom_segment(aes(x=x, y=y, xend=xend, yend=yend, color=color), data=arrow_df,
               arrow=arrow(ends = "first", length = unit(0.1, "npc"), type = "closed"),
               linewidth=0, linejoin = "bevel") +
  geom_segment(aes(x=x, y=y+0.5, xend=xend, yend=yend, color=color),
               data=arrow_df, linewidth=2) +
  geom_text(aes(x=lab_x, y=yend+0.5, color=color, label=cyc_type),
            data=arrow_df, vjust=0, hjust=0, lineheight=0.8) +
  scale_x_discrete(breaks=as.character(seq(4, 14, 2))) +
  scale_fill_gradient2(
    low = scales::muted("blue"), high = scales::muted("red"),  mid = "grey90",
    aesthetics = c("fill"), guide = "none"
  ) +
  scale_color_identity() +
  theme_bw() +
  theme(legend.position = "top", axis.text.x = element_blank(), 
        legend.margin = margin(t = 10,0,0,0), 
        legend.box.margin = margin(-10,-10,-10,-10),
        plot.margin = unit(numeric(4), "cm"),
        strip.text = element_text(face = "bold"),
        plot.background = element_rect(fill="white"))

pcbarplot <- frac_pc_in_metab_df %>% 
  distinct(station, depth, PC_um) %>%
  ggplot(aes(x=station, y=PC_um)) +
  geom_col(fill="grey30") +
  facet_grid(~depth, scales="free") +
  labs(y="Particulate\ncarbon (μM)") +
  scale_x_discrete(breaks=as.character(seq(4, 14, 2))) +
  scale_y_continuous(expand = c(0.005, 0.1)) +
  theme_bw() +
  theme(strip.background = element_blank(), strip.text = element_blank(),
        axis.text.x = element_blank(), axis.title.x = element_blank(),
        plot.margin = margin(0,1,0,0), plot.background = element_rect(fill="white"))
  

med_frac_df <- frac_pc_in_metab_df %>%
  group_by(depth, station, sla_corr) %>%
  summarize(frac_C_in_metab=mean(frac_C_in_metab)) %>%
  ungroup()
pc_frac_metab_gp <- frac_pc_in_metab_df %>%
  ggplot(aes(x=station, y=frac_C_in_metab)) +
  geom_hline(yintercept = 0) +
  # geom_boxplot(aes(color=sla_corr), width = 0.9, linewidth=0.8) +
  geom_point(color="black", size=2) +
  geom_point(aes(fill=sla_corr), data=med_frac_df, size=4, pch=21, alpha=0.7) +
  scale_x_discrete(breaks=as.character(seq(4, 14, 2))) +
  scale_y_continuous(labels = scales::label_percent()) +
  facet_grid(~depth, scales="free") +
  scale_color_gradient2(
    low = scales::muted("blue"), high = scales::muted("red"),  mid = "grey90",
    aesthetics = c("color", "fill")
  ) +
  guides(fill=guide_colorbar(direction = "horizontal", title.position = "top", 
                              title.hjust = 0.5, barheight = 0.2, reverse = TRUE)) +
  labs(fill="Corr. SLA (cm)", x="Station",
       y="Fraction of PC represented\nby known metabolites") +
  theme_bw() +
  theme(legend.position = c(0.84, 0.75), legend.background = element_rect(color="grey90"),
        strip.background = element_blank(), strip.text = element_blank(),
        plot.margin = margin(0,1,0,0), plot.background = element_rect(fill="white"))

# targ_gp_w_sla_frac <- egg::ggarrange(barplot_targ_gp, pc_frac_metab_gp, draw = FALSE)
# ggsave("targ_gp_w_sla_frac.png", plot = targ_gp_w_sla_frac, device = "png", 
#        path = "figures", width = 6.875, height = 4.5, units = "in", dpi = 300, 
#        bg = "white")
targ_gp_w_sla_frac <- egg::ggarrange(barplot_targ_gp, pcbarplot, pc_frac_metab_gp,
                                     draw = FALSE, heights = c(0.4, 0.2, 0.4))
ggsave("fig_3_targ_gp_w_sla_frac.tif", plot = targ_gp_w_sla_frac, device = "tif",
       path = "figures", width = 6.875, height = 5.5, units = "in", dpi = 300,
       bg = "white")
ggsave("fig_3_targ_gp_w_sla_frac.png", plot = targ_gp_w_sla_frac, device = "png",
       path = "figures/lowres", width = 6.875, height = 5.5, units = "in", dpi = 144,
       bg = "white")
```

![](figures/fig_3_targ_gp_w_sla_frac.tif)

*Figure 3: Differences in known metabolite concentration across the pair of adjacent eddies in the MESO-SCOPE transect separated by depth. Top panels depict concentrations of known compounds measured, where bar height corresponds to median triplicate concentration for each metabolite with the top 9 shown and the 44 other identified metabolites summed in grey. TMAO = trimethylamine N-oxide, HO-Ile = hydroxyisoleucine, GBT = glycine betaine, DCM = deep chlorophyll maximum. Center panels show the corresponding measurements of particulate carbon (PC), while the lower panels depict the fraction of total particulate carbon in the known metabolites. The mean value of three biological triplicates is shown in color and the raw values in black behind. Colors correspond to corrected sea level anomaly (Corr. SLA), with dark red indicating anticyclonic (positive SLA) and dark blue indicating cyclonic (negative SLA) eddy state.*

```{r avg targ conc values and numbers for text, eval=FALSE}
avg_targ_concs %>%
  mutate(compound_name=fct_inorder(compound_name)) %>%
  mutate(depth=factor(depth, levels=c("Surface", "DCM", "Deep"))) %>%
  group_by(depth, compound_name) %>%
  summarise(nM=mean(nM)) %>%
  arrange(compound_name) %>%
  pivot_wider(names_from = depth, values_from = nM) %>%
  write_csv("supplement/avg_targ_conc_by_depth.csv")
avg_targ_concs %>%
  mutate(depth=factor(depth, levels=rev(c("Surface", "DCM", "Deep")))) %>%
  group_by(compound_name, depth) %>%
  summarise(nM=mean(nM)) %>%
  ggplot(aes(x=nM, y=depth, group=compound_name)) +
  geom_path() +
  scale_x_log10()
avg_targ_concs %>%
  mutate(depth=factor(depth, levels=rev(c("Surface", "DCM", "Deep")))) %>%
  group_by(compound_name, depth) %>%
  summarise(nM=mean(nM)) %>%
  mutate(nM=rank(nM)) %>%
  ungroup() %>%
  pivot_wider(names_from = depth, values_from = nM) %>%
  mutate(depth_class=paste(Surface, DCM, Deep)) %>%
  count(depth_class)
```

The most abundant intracellular metabolite quantified in the eddy transect samples was trimethylamine N-oxide, with an average concentration of 1.38 nM and a distinct DCM maximum (Figure 3), though a few samples had concentrations an order of magnitude higher for unknown reasons (Supplemental Figure 3). A majority of the known molecules (37/53) had a similar pattern with a subsurface maximum at the DCM, including glycine betaine, glutamate, and guanine. Thirteen molecules obeyed a different pattern and decreased monotonically with depth, including the molecules hydroxyisoleucine, gonyol, and dimethylsulfoniopropionate. Two known molecules (arsenobetaine and O-acetylcarnitine) increased with depth, potentially representing particle degradation during export.

```{r PC vs total_nM ANOVA by depth, eval=FALSE}
# More carbon in metabolites at DCM when outliers removed?
no_outlier_frac_pc_in_metab_df %>%
  mutate(pred_total_nM=predict(lm(formula=total_nM~PC_um, data=.), newdata=.)) %>%
  select(filename, depth, station, sla_class, sla_corr, total_nM, PC_um, pred_total_nM) %>%
  mutate(total_nM_resid=total_nM-pred_total_nM) %>%
  aov(formula=total_nM_resid~depth) %>% TukeyHSD() %>% broom::tidy()
  # ggplot() + geom_boxplot(aes(x=depth, y=total_nM_resid)) + aes(color=sla_class)

# More carbon in metabolites at DCM when outliers handled with rank testing?
frac_pc_in_metab_df %>%
  mutate(pred_total_nM=predict(lm(formula=total_nM~PC_um, data=.), newdata=.)) %>%
  select(filename, depth, total_nM, PC_um, pred_total_nM) %>%
  mutate(total_nM_resid=rank(total_nM-pred_total_nM)) %>%
  kruskal.test(formula=total_nM_resid~depth) %>% broom::tidy()
  # ggplot() + geom_boxplot(aes(x=depth, y=total_nM_resid))

# More carbon in metabolites at DCM when measured by % in metabs too
no_outlier_frac_pc_in_metab_df %>%
  aov(formula=frac_in_metab~depth) %>% TukeyHSD() %>% broom::tidy()
  # ggplot() + geom_boxplot(aes(x=depth, y=frac_in_metab))
no_outlier_frac_pc_in_metab_df %>%
  group_by(depth) %>%
  summarise(median(frac_in_metab))
frac_pc_in_metab_df %>%
  kruskal.test(formula=frac_in_metab~depth, data=.) %>% broom::tidy()
  # ggplot() + geom_boxplot(aes(x=depth, y=rank(frac_in_metab)))
```

```{r comparison to Angies PC values, eval=FALSE}
angie_pcdata <- "../metadata/raw_data/Karl_HL2_WC_PCPN_Final_Data.xlsx" %>%
  openxlsx::read.xlsx() %>%
  slice(-1) %>%
  select(abs_depth=`Depth.(m)`, PC_um=`C.value.(µmol/L)`) %>%
  # filter(abs_depth<30) %>%
  mutate(cruise="HOE-Legacy")

my_pcdata <- filled_file_metadata %>%
  # filter(depth=="Surface") %>%
  distinct(abs_depth, PC_um) %>%
  mutate(cruise="MESO-SCOPE")

rbind(angie_pcdata, my_pcdata) %>%
  ggplot() +
  geom_point(aes(x=PC_um, y=-abs_depth, color=cruise))
```

```{r homoserinebetaine search, eval=FALSE}
library(arrow)
dset <- open_dataset("../tmzMLs/MT_all_file_arr_pqds/")
hmb_eic <- dset %>%
  filter(mz%between%pmppm(162.1125, 10)) %>%
  # filter(mz%between%pmppm(120.0655, 10)) %>%
  collect()
hmb_eic %>%
  filter(rt%between%c(6, 11)) %>%
  # filter(rt%between%c(9, 13)) %>%
  mutate(samp_type=str_extract(filename, "Poo|Smp|InH2O|Mat")) %>%
  # distinct(filename, samp_type) %>%
  # filter(samp_type!="Smp")
  mutate(plotcol=ifelse(str_detect(filename, "180821_Smp_MS11C215m_A.mzML"), 
                          "MS11C215m_A", samp_type)) %>%
  ggplot() +
  geom_line(aes(x=rt, y=int, group=filename, color=plotcol)) +
  facet_wrap(~samp_type, ncol=1, scales="free_y")
untarg_feats %>% 
  filter(cruise=="MS") %>%
  filter(mzmed%between%pmppm(162.1125))
untarg_peaks %>%
  filter(feature%in%c("MS_POS_1246", "MS_POS_1247", "MS_POS_1248")) %>%
  mutate(feature=factor(feature, levels=c("MS_POS_1248", "MS_POS_1246", "MS_POS_1247"),
                        labels=c("Threonine betaine (RT=7.1)", 
                                 "Homoserine betaine (RT=8.4)",
                                 "Carnitine (RT=9.9)"))) %>%
  left_join(filled_file_metadata) %>%
  filter(ms_run=="MT") %>%
  mutate(station=factor(station, levels=as.character(4:14))) %>%
  mutate(depth=factor(depth, levels=c("Surface", "DCM", "Deep"))) %>%
  mutate(plotcol=ifelse(filename=="180821_Smp_MS11C215m_A.mzML", "red", "black")) %>%
  ggplot(aes(x=station, y=norm_area, label=filename, color=plotcol)) +
  geom_point() +
  facet_grid(feature~depth, scales="free_y") +
  geom_hline(yintercept = 0) +
  scale_color_identity()
untarg_peaks %>%
  filter(feature%in%c("MS_POS_1246", "MS_POS_1247", "MS_POS_1248")) %>%
  left_join(filled_file_metadata) %>%
  filter(ms_run=="MT") %>%
  group_by(feature) %>%
  mutate(norm_area=norm_area/norm_area[filename=="180821_Smp_MS11C215m_A.mzML"]) %>%
  group_by(feature, depth) %>%
  summarise(mean_area=max(norm_area)) %>%
  pivot_wider(names_from = "depth", values_from = mean_area)
```

We also measured several phosphorus-containing compounds and their non-phosphorus equivalents (Supplemental Figure 4A). Phosphocholine is the polar headgroup for phosphatidylcholine, a class of intact polar diacylglycerols that has been previously described at this study site as showing phosphorus stress relief among eukaryotic organisms in response to the upwelling of phosphate-rich deep sea water [@Bent2024]. We measured this metabolite as well as choline (the non-phosphate containing equivalent) and glycerophosphocholine (GPC, which contains an additional glycerol group) with the expectation that phosphorus stress would result in higher choline:phosphocholine and choline:GPC ratios, as shown in Seelen et al. (in review). However, we saw no differences in this ratio across the eddy transect after accounting for depth effects (Supplemental Figure 4C). Additionally, choline:GPC ratios were highest at 175m where PO$_4^{3-}$ concentrations are highest, indicating that this metric may not accurately measure P-stress. Other phosphorus-containing lipid headgroups glycerophosphoglycerol and glycerophosphoethanolamine also showed no trend across the eddy at a given depth and instead showed trends functionally identical to the average metabolite in Figure 2 (Supplemental Figure 4B).

## Particulate organic matter compositional shifts across eddy transect

In addition to absolute shifts in metabolite concentration due to biomass differences, we also explored compositional shifts in the metabolomes by assessing the fraction of the total metabolite pool contributed by each compound across the eddy transect. This allowed us to isolate the signal due to shifts in community composition and organismal response from those introduced by changes in biomass of the community as a whole.

```{r regression relative abundance coefficients, eval=FALSE}
reg_coefs <- untarg_peaks %>%
  # filter(feature!="MS_NEG_0155") %>%
  left_join(filled_file_metadata %>% distinct(filename, ms_run, station, depth, sla_corr)) %>%
  filter(ms_run=="MT") %>%
  filter(depth%in%c("Surface", "DCM", "Deep")) %>%
  filter(feature%in%unique(feature)) %>%
  group_by(filename) %>%
  mutate(norm_area=norm_area/sum(norm_area)) %>%
  ungroup() %>%
  nest(data=-c(depth, feature)) %>%
  mutate(test=map(data, ~broom::tidy(lm(.x$norm_area~.x$sla_corr)), .progress = TRUE)) %>%
  unnest(test)
reg_coefs %>%
  filter(str_detect(term, "sla_corr")) %>%
  mutate(p.value=p.adjust(p.value, method="fdr")) %>%
  arrange(p.value) %>%
  left_join(untarg_feats) %>%
  select(feature, depth, estimate, p.value, mzmed, rtmed, compound_name) %>%
  filter(p.value<0.05) %>%
  # as.data.frame() %>%
  filter(depth=="Deep") %>% 
  print(n=Inf)
reg_coefs %>%
  filter(str_detect(term, "sla_corr")) %>%
  left_join(untarg_feats) %>%
  filter(compound_name=="Homarine") %>%
  # filter(feature=="MS_POS_1395") %>%
  filter(depth=="DCM") %>%
  select(compound_name, depth, data, estimate) %>%
  unnest(data) %>%
  group_by(sla_corr, station) %>%
  ggplot(aes(x=sla_corr, y=norm_area*100)) +
  geom_smooth(method="lm", formula="y~x") +
  geom_boxplot(aes(group=sla_corr)) +
  scale_y_continuous(limits = c(0, NA))
```

In the 15 meter surface samples, only seven compounds had significant changes in relative peak area across the eddy transect, two of which were known: 5-oxoproline, which decreased from 0.2-0.3% of the total peak area at the center of the anticyclone to 0.05-0.1% in the cyclone; and a combined peak of sarcosine and beta-alanine which increased from 0.01% to 0.02-0.03%. The largest shift significant at an α = 0.05 level was an unknown mass feature with *m/z* = 189.12338 and a retention time around 8.6 minutes, the relative contribution of which increased from approximately 0.2-0.3% in the cyclone to ~1.5% in the anticyclone.

At the DCM, twenty-two compounds changed significantly in relative peak area across the transect. Seven of these were known metabolites, of which trigonelline and homarine were most enriched in the cyclone while arsenobetaine was the only metabolite with a larger fraction of the total peak area in the anticyclone. Homarine showed a very large and highly significant shift, representing about 2.5% of the total peak area in the anticyclone but 7-8% in the cyclone center. Trigonelline (N-methyl niacin), an isomer structurally very similar to homarine but biologically distinct, had a surprisingly strong correlation with homarine (Pearson's $r=0.855$). Its peak areas were consistently around one third of homarine's but still large enough to have the second-largest shift in relative peak area of the significantly different compounds. Similarly, arsenobetaine represented about 0.1% of the peak area in the cyclone and 0.5-0.6% of the total peak area in the anticyclone. The most significantly different compounds at the DCM in each direction, however, were both unknowns. The lowest p-value ($6.4 × 10^{-6}$ after FDR correction) in the DCM data was a mass feature with an *m/z* of 173.09211, a retention time also around 8.6 minutes, and enrichment in the anticyclone with a putative chemical formula of [M+H] = C$_7$H$_{13}$N$_2$O$_3$, possibly glycylproline or prolylglycine. The next-lowest (p = $2.5 × 10^{-4}$ after FDR correction) was enriched in the cyclone and had an *m/z* of 275.0712 and a retention time of 11.4 minutes with a putative formula of C$_{18}$H$_{11}$O$_3$.

Finally, in the 175 meter samples we detected 43 mass features with a significant association with SLA. Notably, all detected nucleobases (guanine, adenine, and cytosine) were positively associated with SLA (higher values in the anticyclone than the cyclone), though the strongest associations were found in O-acetylcarnitine, betonicine, and tyrosine. Both O-acetylcarnitine and betonicine effectively quintupled their contribution to total peak area, shifting from approximately 0.1% to ~0.5% over the transect. Acetylcholine was the only known compound more relatively abundant in the cyclone than in the anticyclone along with one unknown of *m/z* 131.0340. The unknown at *m/z* 173.0921 with the strongest trend at the DCM again had the strongest trend in the 175 meter samples, here with an FDR-corrected p-value of $2.0 × 10^{-7}$.

```{r homarine/trigonelline correlation, eval=FALSE}
reg_coefs %>%
  filter(str_detect(term, "sla_corr")) %>%
  left_join(untarg_feats) %>%
  filter(compound_name%in%c("Homarine", "Trigonelline")) %>%
  filter(depth=="DCM") %>%
  select(compound_name, depth, data, estimate) %>%
  unnest(data) %>%
  select(compound_name, filename, norm_area) %>%
  pivot_wider(names_from=compound_name, values_from=norm_area) %>%
  column_to_rownames("filename") %>%
  cor()
```



## High vertical resolution sampling near eddy center DCM reveals distinct metabolome between cyclone and anticyclone

```{r MC NMDS and PERMANOVA calculations}
set.seed(123)
multivar_analysis_MC <- untarg_peaks %>%
  left_join(filled_file_metadata %>% distinct(ms_run, station, filename, depth), 
          by="filename") %>% filter(ms_run=="MC") %>% 
  mutate(depth=factor(depth, levels=c("DCMless20m", "DCMless10m", "DCM", 
                                      "DCMplus10m", "DCMplus20m"))) %>%
  nest(data=everything()) %>%
  mutate(wide_peaks=map(data, function(metab_df){
    metab_df %>%
      select(feature, filename, norm_area) %>%
      group_by(feature) %>%
      filter(sum(norm_area==0)<n()*0.9) %>%
      mutate(norm_area=scale(norm_area)[,1]) %>%
      pivot_wider(names_from = "feature", values_from = norm_area) %>%
      column_to_rownames("filename") %>%
      data.matrix()
  })) %>%
  mutate(nmds_output=map(wide_peaks, function(metab_mat){
    env_vec <- pull(filled_file_metadata, sla_corr, filename)[rownames(metab_mat)]
    vegan::metaMDS(metab_mat, k = 2, distance = "manhattan", trace = 0, 
                   autotransform = FALSE, noshare = FALSE, wascores = FALSE) %>%
    vegan::MDSrotate(env_vec)
  })) %>%
  mutate(nmds_stress=map_dbl(nmds_output, pluck, "stress")) %>%
  mutate(perm_output=map(wide_peaks, function(metab_mat){
    adon_expl <- data.frame(filename=rownames(metab_mat)) %>%
      left_join(filled_file_metadata %>% select(filename, sla_class, depth), by="filename")
    vegan::adonis2(
      metab_mat~sla_class+depth, data = adon_expl, by = "margin", 
      permutations = 999, method = "manhattan")
  })) %>%
  mutate(perm_r2_sla=map_dbl(perm_output, pluck, "R2", 1)) %>%
  mutate(perm_r2_depth=map_dbl(perm_output, pluck, "R2", 2)) %>%
  mutate(perm_pval=map_dbl(perm_output, pluck, "Pr(>F)", 1)) %>%
  mutate(pca_output=map(wide_peaks, prcomp)) %>%
  mutate(nmds_stress=paste("NMDS stress =", round(nmds_stress, digits = 3)))
  # mutate(perm_r2=round(perm_r2, digits = 2)) %>%
  # mutate(perm_pval=round(perm_pval, digits = 4)) %>%
  # mutate(perm_label=paste0("R2 = ", perm_r2, "\np = ", perm_pval))
```

We further investigated the metabolomic response to SLA at the DCM by collecting samples taken at high-resolution depth intervals around the DCM at the center of both eddy poles. Here, we found that this ~40 meter depth range and the transition between the eddies explained similar amounts of variation in the data (R$^2_{depth}=0.253$, R$^2_{SLA}=0.238$) and were both highly significant factors (permutational p-values << 0.001, Figure 4). Additionally, both sample depth and SLA were correlated with the first principal component of the metabolite matrix ($r_{depth}=0.766$, $r_{SLA}=0.684$, fraction of variance explained by PC1 = 30.8%). The cyclonic samples in particular show a much larger spread than the anticyclonic ones, indicating larger sample variability in the cyclone DCM relative to the anticyclone. A depth gradient is visible along NMDS 2, with the deepest samples generally at the top right of the plot and the shallowest ones closer to the bottom (Figure 4).

```{r figure MC NMDS plot}
MC_nmds_gp <- multivar_analysis_MC$nmds_output %>%
  pluck(1, "points") %>% 
  as.data.frame() %>%
  rownames_to_column("filename") %>%
  left_join(filled_file_metadata %>% distinct(filename, sla_class, depth, tripl)) %>%
  mutate(depth=factor(depth, levels=c("DCMless20m", "DCMless10m", "DCM", 
                                      "DCMplus10m", "DCMplus20m"),
                      labels=c("-20m", "-10m", "DCM", "+10m", "+20m"))) %>%
  mutate(fill_col=case_when(
    depth=="-20m" & sla_class=="Cyclone"~"#c4c4e0",
    depth=="-10m" & sla_class=="Cyclone"~"#8989c1",
    depth=="DCM" & sla_class=="Cyclone"~"#6161ad",
    depth=="+10m" & sla_class=="Cyclone"~"#343489",
    depth=="+20m" & sla_class=="Cyclone"~"#23235b",
    depth=="-20m" & sla_class=="Anticyclone"~"#dabdbd",
    depth=="-10m" & sla_class=="Anticyclone"~"#b57c7c",
    depth=="DCM" & sla_class=="Anticyclone"~"#9c5050",
    depth=="+10m" & sla_class=="Anticyclone"~"#762020",
    depth=="+20m" & sla_class=="Anticyclone"~"#4f1616"
  )) %>%
  ggplot() +
  geom_vline(xintercept = 0) +
  geom_hline(yintercept = 0) +
  geom_point(aes(x=MDS1, y=MDS2, fill=fill_col, shape=sla_class, color=depth), size=6) +
  scale_fill_identity() +
  scale_color_manual(values = rep("white", 5)) +
  scale_shape_manual(breaks=c("Cyclone", "Anticyclone"), values=c(25, 24)) +
  annotate("label", x=-200, y=100, label="Cyclone", color=scales::muted("blue"),
           label.r = unit(0, "in"), size=6) +
  annotate("label", x=100, y=120, label="Anticyclone", color=scales::muted("red"),
           label.r = unit(0, "in"), size=6) +
  guides(alpha="none", fill="none", shape="none", color=guide_legend(
    title = "Depth:    ", title.position = "left", title.hjust = 0.5, direction = "horizontal",
    keywidth=0.2, keyheight=0.1,
    override.aes = list(fill=c("grey90", "grey80", "grey60", "grey40", "grey20"), 
                        pch=21, color="black")
  )) +
  labs(x="NMDS axis 1", y="NMDS axis 2") +
  theme_bw() +
  theme(legend.position = "top", legend.text = element_text(margin = margin(r = 15, l=2, unit = "pt"))) +
  annotate("label", x=-Inf, y=Inf, hjust=0, vjust=1, label.r = unit(0, "in"),
           label=expression(atop(paste("R"[SLA]^2, "=0.238"), paste("R"[depth]^2, "=0.253")))) +
  annotate("label", x=-Inf, y=-Inf, label=multivar_analysis_MC$nmds_stress, 
           hjust=0, vjust=-2, label.r = unit(0, "in"))

ggsave("fig_4_MC_nmds_gp.tif", plot = MC_nmds_gp, device = "tif", 
       path = "figures", width = 6.875, height = 4.5, units = "in", dpi = 300)
ggsave("fig_4_MC_nmds_gp.png", plot = MC_nmds_gp, device = "png", 
       path = "figures/lowres", width = 6.875, height = 4.5, units = "in", dpi = 144)
```

![](figures/fig_4_MC_nmds_gp.tif)

*Figure 4: NMDS plot of high-resolution depth sampling around the deep chlorophyll maximum (DCM, ~115 meters) at the two eddy centers during MESO-SCOPE. Red upward-pointed triangles are from the anticyclone and blue downward-pointed ones are from the cyclone. Shading intensity reflects the depth above or below the DCM. PERMANOVA estimates of the variance explained by depth and sea level anomaly (SLA) are noted in the upper left corner and the NMDS stress value is reported in the bottom left.*

```{r PCA checks for MC axis regressions, eval=FALSE}
multivar_analysis_MC$pca_output[[1]]$x %>%
  as.data.frame() %>%
  rownames_to_column("filename") %>%
  left_join(filled_file_metadata) %>%
  select(abs_depth, sla_corr, PC1:PC3) %>%
  # cor() %>% round(3)
  ggplot(aes(x=abs_depth, y=PC1, color=sla_corr)) +
  geom_point()

head(multivar_analysis_MC$pca_output[[1]]$sdev^2/sum(multivar_analysis_MC$pca_output[[1]]$sdev^2)*100)
```

To characterize the observed differences in multivariate space we used k-means clustering as an unsupervised way to identify dominant trends within the data (Figure 5). We found that a majority of the metabolites (Clusters 1 and 2, 63% of the total) fell into clusters with larger peak areas in the cyclone, as expected, while also detecting 32 metabolites that clustered such that the mean metabolite was enriched in the anticyclone (Cluster 4, Figure 5). Cluster 2 had a distinct decrease in relative peak area with depth, while the other clusters had much less clear depth trends. Cluster 1 also showed a DCM maximum for the samples from the anticyclone, correlating well with flow cytometry counts of picoeukaryotes (Pearson's $r=0.851$) that was not present in the cyclone ($r=0.013$).

```{r k-means clusterplot}
set.seed(123)
kclust <- untarg_peaks %>%
  left_join(filled_file_metadata, by="filename") %>%
  filter(ms_run=="MC") %>%
  select(feature, filename, norm_area) %>%
  group_by(feature) %>%
  filter(sum(norm_area==0)<n()*0.9) %>%
  mutate(norm_area=scale(norm_area)[,1]) %>%
  pivot_wider(names_from = "filename", values_from = norm_area) %>%
  column_to_rownames("feature") %>%
  data.matrix() %>%
  kmeans(centers = 4)

kclust_nms <- names(kclust$cluster)
kclust$cluster <- order(kclust$size, decreasing = TRUE)[kclust$cluster]
names(kclust$cluster) <- kclust_nms
kclust$size <- sort(kclust$size, decreasing = TRUE)

kclust_df <- kclust %>%
  pluck("cluster") %>%
  data.frame(cluster=.) %>%
  rownames_to_column("feature") %>%
  left_join(untarg_peaks, multiple = "all", by="feature") %>%
  left_join(filled_file_metadata, by="filename") %>%
  filter(ms_run=="MC") %>%
  arrange(desc(abs_depth)) %>%
  mutate(depth=fct_inorder(depth)) %>%
  group_by(feature) %>%
  mutate(norm_area=scale(norm_area)[,1]) %>%
  group_by(cluster, depth, sla_class) %>%
  summarise(mean_rank=mean(norm_area), iqr_value=sd(norm_area), .groups = "drop")

lab_df <- tribble(
  ~xlab, ~ylab, ~sla_class, ~clusterlab,
  0, 5.5, "Anticyclone", "Cluster #1: n=73",
  0.8, 1, "Cyclone", "Cluster #1: n=73"
)

library(ggh4x)
strip_theme <- strip_themed(background_x = elem_list_rect(fill= c(
  "#414487FF", "#7AD151FF", "#22A884FF", "#2A788EFF")))
kclust_gp <- kclust_df %>%
  left_join(data.frame(size=kclust$size) %>% mutate(cluster=row_number()), by="cluster") %>%
  mutate(clusterlab=paste0("Cluster #", cluster, ": n=", size)) %>%
  mutate(depth=factor(depth, labels=rev(c("-20m", "-10m", "DCM", "+10m", "+20m")),
                      levels=rev(c("DCMless20m", "DCMless10m", "DCM", "DCMplus10m", "DCMplus20m")))) %>%
  ggplot(aes(y=depth, color=sla_class)) +
  geom_vline(xintercept = 0) +
  geom_path(aes(x=mean_rank, group=sla_class), linewidth=1) +
  geom_point(aes(x=mean_rank), size=3) +
  geom_label(aes(x=xlab, y=ylab, label=sla_class, color=sla_class), 
             data=lab_df, label.r = unit(0, "in")) +
  ggh4x::facet_wrap2(~clusterlab, nrow = 1, strip = strip_theme) +
  scale_color_manual(values = c(scales::muted("blue"), scales::muted("red"), "grey50"), 
                     breaks = c("Cyclone", "Anticyclone", "Neither"),
                     guide="none") +
  labs(x="Mean z-scaled metabolite area", y="Sampling depth", color=NULL) +
  coord_cartesian(ylim = c(1, 5.5)) +
  theme_bw() +
  theme(plot.background = element_rect(fill = "white"),
        strip.background = element_rect(fill = "white"),
        strip.text = element_text(face = "bold", color="white"))
```

```{r pairwise_tests and fold_change_df}
MC_simpledata <- untarg_peaks %>%
  left_join(filled_file_metadata %>% distinct(filename, ms_run, station), by="filename") %>%
  filter(ms_run=="MC") %>%
  select(feature, filename, norm_area, station) %>%
  group_by(feature) %>%
  filter(sum(norm_area==0)<n()*0.9) %>%
  group_by(filename) %>%
  # mutate(norm_area=norm_area/sum(norm_area)) %>%
  ungroup()
pairwise_tests <- MC_simpledata %>%
  nest(data=c(filename, norm_area, station)) %>%
  mutate(stattest=map(data, ~broom::tidy(wilcox.test(.x$norm_area~.x$station)))) %>%
  unnest(stattest) %>%
  select(feature, statistic, p.value) %>%
  mutate(p.value=p.adjust(p.value, method="fdr"))

fold_change_df <- MC_simpledata %>%
  group_by(feature, station) %>%
  summarise(mean_area=mean(norm_area, na.rm=TRUE)) %>%
  pivot_wider(names_from=station, values_from=mean_area) %>%
  mutate(fold_change=L1/L2) %>%
  select(feature, fold_change)
```

```{r eddy center volcano plot}
label_df <- pairwise_tests %>%
  left_join(untarg_feats) %>%
  filter(compound_name%in%c("Isethionic acid", "Trigonelline", "Arsenobetaine",
                            "Homarine", "Dimethylsulfonioacetate", "Dimethylsulfoniopropionate")) %>%
  left_join(fold_change_df, by = join_by(feature)) %>%
  mutate(xspot = log2(fold_change), yspot=-log10(p.value))

volcanoplot_gp <- pairwise_tests %>%
  left_join(fold_change_df, by = join_by(feature)) %>%
  left_join(rownames_to_column(data.frame(cluster=kclust$cluster), "feature"), 
            by = join_by(feature)) %>%
  left_join(untarg_feats %>% select(feature, compound_name), by = join_by(feature)) %>%
  mutate(known=ifelse(is.na(compound_name), "Unknown", "Known")) %>%
  mutate(compound_name=ifelse(is.na(compound_name), feature, compound_name)) %>%
  ggplot(aes(x=log2(fold_change), y=-log10(p.value), color=factor(cluster), 
             shape=known, label=compound_name)) +
  geom_vline(xintercept = 0) +
  geom_hline(yintercept = 0) +
  geom_hline(yintercept = 1.3, lty=2) +
  annotate("segment", x=label_df$xspot[5], y=6.8, xend=label_df$xspot[5], yend=label_df$yspot[5], 
           # arrow=arrow(type="closed", angle = 25, length = unit(0.05, "npc")), 
           lwd=0.7, color="grey50") +
  annotate("label", x=-2, y=6.8, label="Arsenobetaine", hjust=0, label.r=unit(0, "in")) +
  annotate("segment", x=1.5, y=6.8, xend=label_df$xspot[3], yend=label_df$yspot[3], 
           lwd=0.7, color="grey50") +
  annotate("label", x=1.5, y=6.8, label="Homarine", hjust=1, label.r=unit(0, "in")) +
  annotate("segment", x=2, y=6.8, xend=label_df$xspot[2], yend=label_df$yspot[2], 
           lwd=0.7, color="grey50") +
  annotate("label", x=2, y=6.8, label="DMSP", hjust=0, label.r=unit(0, "in")) +
  annotate("segment", x=Inf, y=label_df$yspot[1], xend=label_df$xspot[1], yend=label_df$yspot[1], 
           lwd=0.7, color="grey50") +
  annotate("label", x=Inf, y=label_df$yspot[1], label="DMS-Ac", hjust=1, vjust=0.3, label.r=unit(0, "in")) +
  annotate("segment", x=3.5, y=5, xend=1.9, yend=5, 
           lwd=0.7, color="grey50") +
  annotate("segment", x=1.9, y=5, xend=label_df$xspot[4], yend=label_df$yspot[4], 
           lwd=0.7, color="grey50") +
  annotate("label", x=Inf, y=5, label="Trigonelline", hjust=1, label.r=unit(0, "in")) +
  annotate("segment", x=3.6, y=3, xend=2.2, yend=3, 
           lwd=0.7, color="grey50") +
  annotate("segment", x=2.2, y=3, xend=label_df$xspot[6], yend=label_df$yspot[6], 
           lwd=0.7, color="grey50") +
  annotate("label", x=Inf, y=3, label="Isethionate", hjust=1, label.r=unit(0, "in")) +
  geom_point(size=2, stroke=1) +
  scale_x_continuous(breaks=c(-4:4)) +
  scale_y_continuous(breaks = 0:6, labels = paste0("10<sup>-", 0:6, "</sup>"), expand = expansion(c(0.01, 0.05))) +
  scale_shape_manual(values=c(4, 16), breaks = c("Unknown", "Known")) +
  scale_color_manual(
    values = c("#414487FF", "#7AD151FF", "#22A884FF", "#2A788EFF"),
    breaks = 1:4
  ) +
  theme_bw() +
  labs(x="Log-2 fold-change cyclone enrichment", y="Mann-Whitney p-value\n(FDR-corrected)",
       shape="Metabolite status", color="k-means\ncluster") +
  guides(color="none", shape=guide_legend(order=1)) +
  theme(axis.text.y = ggtext::element_markdown(),
        plot.background = element_rect(fill = "white"),
        legend.position = c(0, 1), legend.justification = c(0, 1), 
        legend.background = element_rect(fill="#FFFFFF88", color="black"),
        legend.spacing.y=unit(0, "in"),
        legend.key.size = unit(0, "cm"))
```

```{r kclust volcano plot combine}
kclust_volcano_gp <- egg::ggarrange(
  kclust_gp, volcanoplot_gp, 
  ncol=1, heights = c(1.5, 2), draw = FALSE)
ggsave("fig_5_kclust_volcano_gp.tif", plot = kclust_volcano_gp, device = "tif", 
       path = "figures", width = 6.875, height = 5, units = "in", dpi = 300,
       bg="white")
ggsave("fig_5_kclust_volcano_gp.png", plot = kclust_volcano_gp, device = "png", 
       path = "figures/lowres", width = 6.875, height = 5, units = "in", dpi = 144,
       bg="white")
```

![](figures/fig_5_kclust_volcano_gp.tif)

*Figure 5: Distribution of metabolites in the high-resolution depth samples from the centers of each MESO-SCOPE eddy. The upper row of plots shows k-means clusters where points denote the average z-scored peak area for both known and unknown metabolites across the samples and are colored by the eddy from which they were taken. Clusters have been ordered by number of metabolites in each group and the total is denoted in the panel titles. Both depth trends (mostly a net decrease in metabolites with depth) and eddy effects (cyclonic enrichment in clusters 1 and 2, anticyclone enrichment in cluster 4) are observable. The lower plot shows the individual known and unknown metabolites where points correspond to the FDR-corrected p-value estimated by the nonparametric Mann-Whitney U test and the log$_2$ fold-change calculated with the average peak area in the cyclone divided by the average peak area in the anticyclone. Colors have been assigned using the k-means clusters and shapes have been assigned based on the status of the mass feature as either a known metabolite that was matched to an authentic standard or an unidentified metabolite. The dashed line across the figure represents the 0.05 level of significance as a visual cue for metabolites above which the differences between the eddies are unlikely to be due to chance. DMSP = dimethylsulfoniopropionate, DMS-Ac = dimethylsulfonioacetate.*

```{r cluster cor w metadata, eval=FALSE}
filled_file_metadata %>% 
  filter(ms_run=="MC") %>% 
  distinct(depth, euk_abund, sla_class) %>% 
  left_join(kclust_df) %>%
  filter(cluster==1) %>%
  filter(sla_class=="Cyclone") %>%
  # with(cor(mean_rank, euk_abund))
  ggplot() +
  geom_point(aes(x=euk_abund, y=mean_rank, color=sla_class))
```

The majority of the individual metabolites in these high-resolution depth profiles differed significantly between the two eddies (121/228, α = 0.05). Of those, 15 were enriched in the anticyclone and 106 were more abundant in the cyclone. As expected, all compounds enriched in the anticyclone were part of Cluster 4 and all those enriched in the cyclone belonged to either Cluster 1 or 2 (Figure 5).

Many of the known metabolites enriched in the cyclone function as osmolytes in the cell and might be enriched due to the increased eukaryotic phytoplankton biomass. However, some metabolites such as isethionate, the reduced sulfur osmolytes dimethylsulfoniopropionate (DMSP) and dimethylsulfonioacetate (DMS-Ac), and the isomers homarine and trigonelline, were enriched in excess of biomass (Figure 5). A few metabolites more abundant in the anticyclone were also given putative identifications based on RT and *m/z* matching with internal standards run at a later time on the mass spectrometer. Of these, the putative arsenobetaine was the most significantly different among these with a peak area at the anticyclone DCM nearly quadruple that of the cyclone (Figure 5). Of note, the 173.0921 *m/z* mass feature noted above as strongly enriched in the anticyclone was among the most significantly different between the two eddies, along with two isomers at an *m/z* of 170.1176 and retention times around 8-9 minutes that increased by a factor of 1.5 in the anticyclone (putative formula C$_7$H$_{14}$N$_4$O).

```{r putative 170.11756 formula, eval=FALSE}
# Two possible formula: C9H16NO2 and C7H14N4O
library(arrow)
dset <- open_dataset("G:/My Drive/AllMeso/tmzMLs/MC_all_file_arr_pqds")

# Too low-intensity for reliable isotope info
dset %>%
  filter(mz%between%pmppm(170.11757, 5)) %>%
  filter(rt%between%c(7, 10)) %>%
  collect() %>%
  left_join(filled_file_metadata) %>%
  ggplot() +
  geom_line(aes(x=rt, y=int, group=filename, color=station))
rtr <- c(8.7, 9.2)
gly_data <- dset %>%
  filter(rt%between%rtr) %>%
  collect()
mzr <- pmppm(170.11756, 10)
reduce(list(
  gly_data[mz%between%mzr] %>%
    group_by(filename) %>%
    summarise(M=sum(int)),
  gly_data[mz%between%(mzr+1.003355)] %>%
    group_by(filename) %>%
    summarise(C13=sum(int))
), merge, all.x=TRUE) %>%
  pivot_longer(C13) %>%
  mutate(prop=value/M) %>%
  print(n=Inf)

# But there are useful MS2 fragments
library(RaMS)
mz_group <- function(mz_vals, ppm){
  group_vec <- numeric(length(mz_vals))
  group_num <- 1L
  init_vec <- mz_vals
  names(init_vec) <- seq_along(init_vec)
  while(length(init_vec)>0){
    mz_i <- init_vec[1]
    err <- mz_i*ppm/1000000
    mz_idxs <- init_vec>mz_i-err & init_vec<mz_i+err
    group_vec[as.numeric(names(mz_idxs)[mz_idxs])] <- group_num
    init_vec <- init_vec[!mz_idxs]
    group_num <- group_num+1L
  }
  group_vec
}
msmsdata_MC <- "../mzMLs/pos/MSMS" %>%
  list.files(, full.names = TRUE, pattern = "180205") %>%
  grabMSdata()
msmsdata_MC$MS2[premz%between%pmppm(170.11757, 10)] %>%
  mutate(mz_group=mz_group(fragmz, 5)) %>%
  group_by(mz_group, filename) %>%
  summarize(fragmz=mean(fragmz), int=mean(int)) %>%
  filter(n()>1) %>%
  group_by(mz_group) %>%
  summarise(fragmz=mean(fragmz), int=mean(int)) %>%
  filter(fragmz<170) %>%
  arrange(desc(int))
# Top two frags have formulae of 125.1075 = C7H13N2
# and 89.07148 = C3H9N2O, both requiring 2 N atoms
```

## Hawaiian Eddy Experiment data reveals a different community and response to mesoscale eddies

Given the strong signals detected in the samples from the MESO-SCOPE cruise both across the eddy transect as well as in the high-resolution DCM sampling, we expected to find similar results in an analogous dataset. Samples were collected during a 2018 cruise on the R/V *Falkor* that again targeted both a cyclonic and an anticyclonic eddy in the North Pacific Subtropical Gyre near Station ALOHA as part of the Hawaiian Eddy Experiment (HEE).

```{r falkor nmds and permanova}
set.seed(123)
scaled_fk_data <- untarg_peaks %>%
  # filter(filename!="190715_Smp_FK180310S62C1-DCM_B.mzML") %>%
  left_join(filled_file_metadata %>% distinct(ms_run, station, filename, depth), 
          by="filename") %>%
  filter(ms_run=="FK") %>%
  mutate(depth=factor(depth, levels=c("Surface", "DCM"))) %>%
  select(feature, filename, norm_area) %>%
  group_by(feature) %>%
  filter(sum(norm_area==0)<n()*0.9)
metab_mat_fk <- scaled_fk_data %>%
  mutate(norm_area=scale(norm_area)[,1]) %>%
  # mutate(norm_area=rank(norm_area)) %>%
  pivot_wider(names_from = "feature", values_from = norm_area) %>%
  column_to_rownames("filename") %>%
  data.matrix()
env_vec <- pull(filled_file_metadata, sla_corr, filename)[rownames(metab_mat_fk)]
nmds_fk <- metab_mat_fk %>%
  vegan::metaMDS(k = 2, distance = "manhattan", trace = 0, autotransform = FALSE,
                 noshare = FALSE, wascores = FALSE) %>%
  vegan::MDSrotate(env_vec)
adon_expl <- data.frame(filename=rownames(metab_mat_fk)) %>%
  left_join(filled_file_metadata %>% select(filename, sla_class, time_class, depth), by="filename")
adon_fk <- vegan::adonis2(
  metab_mat_fk~sla_class+depth, data = adon_expl, by = "margin", 
  permutations = 999, method = "manhattan")
# vegan::adonis2(
#   metab_mat_fk~sla_class+depth+time_class, data = adon_expl, by = "margin",
#   permutations = 999, method = "manhattan")
```

```{r figure falkor nmds}
rot_mat <- matrix(c(sqrt(2)/2, sqrt(2)/2, -sqrt(2)/2, sqrt(2)/2), ncol=2)

fk_nmdsplot <- nmds_fk$points%*%rot_mat %>%
  as.data.frame() %>%
  setNames(c("MDS1", "MDS2")) %>%
  rownames_to_column("filename") %>%
  left_join(filled_file_metadata %>% distinct(filename, sla_class, depth)) %>%
  mutate(MDS1=MDS1*-1) %>%
  mutate(MDS2=MDS2*-1) %>%
  mutate(fill_col=case_when(
    depth=="Surface" & sla_class=="Cyclone"~"25 m Cyclone",
    depth=="DCM" & sla_class=="Cyclone"~"25 m Anticyclone",
    depth=="Surface" & sla_class=="Anticyclone"~"DCM Cyclone",
    depth=="DCM" & sla_class=="Anticyclone"~"DCM Anticyclone"
  )) %>%
  ggplot(aes(label=filename)) +
  geom_vline(xintercept = 0) +
  geom_hline(yintercept = 0) +
  geom_point(aes(x=MDS1, y=MDS2, fill=fill_col, shape=sla_class), size=6, color="white") +
  scale_fill_manual(breaks=c("25 m Cyclone", "25 m Anticyclone", "DCM Cyclone", "DCM Anticyclone"),
                    values=c("#8989c1", "#343489", "#b57c7c", "#762020")) +
  scale_shape_manual(breaks=c("Cyclone", "Anticyclone"), values=c(25, 24)) +
  labs(color="Corr.\nSLA\n(cm)", shape="Depth", x="NMDS axis 1", y="NMDS axis 2") +
  theme_bw() +
  annotate("label", x=-Inf, y=-Inf, hjust=0, vjust=0, label.r = unit(0, "in"),
           label=paste("Stress =", round(nmds_fk$stress, digits = 3))) +
  annotate("label", x=-Inf, y=Inf, hjust=0, vjust=1, label.r = unit(0, "in"),
           label=expression(atop(paste("R"[SLA]^2, "=0.090, p=0.017"),
                                 paste("R"[depth]^2, "=0.251, p<0.001")))) +
  guides(fill=guide_legend(title = NULL, override.aes = list(
    fill=c("#8989c1", "#b57c7c", "#343489", "#762020"), color="white",
    shape=c(25, 24, 25, 24)
  )), shape="none") +
  theme(legend.key.spacing.y = unit(6, "pt")) +
  coord_fixed()

ggsave("fig_6_fk_nmdsplot.tif", plot = fk_nmdsplot, device = "tif",
       path = "figures", width = 6.875, height = 4, units = "in", dpi = 300,
       bg="white")
ggsave("fig_6_fk_nmdsplot.png", plot = fk_nmdsplot, device = "png",
       path = "figures/lowres", width = 6.875, height = 4, units = "in", dpi = 144,
       bg="white")
```

The HEE data was characterized by high inter-replicate variability relative to the samples from the MESO-SCOPE cruise, with 25 meter samples in particular highly variable in multivariate space (Figure 6). Despite this, we still saw the importance of sea level anomaly as a significant explanatory factor in the dataset (PERMANOVA R$^2_{SLA} = 0.090$, p-value = 0.017) as well as the larger depth differences between the surface (25 meters) and DCM (110 - 120 meters) (PERMANOVA R$^2_{depth} = 0.251$, p-value << 0.001). We saw no difference between the 6 AM and 6 PM samples (PERMANOVA R$^2_{time} = 0.031$, p-value = 0.367) despite a few known compounds (trehalose and sucrose) showing large differences, likely due to a majority of the mass features demonstrating no diel effect (Supplemental Figure 5).

![](figures/fig_6_fk_nmdsplot.tif)

*Figure 6: Non-metric multidimensional scaling (NMDS) plot from the Hawaiian Eddy Experiment cruise data, in which points correspond to individual samples and have been colored and shaped by their source eddy status and shaded by the depth from which they were collected. The NMDS stress value has been reported in the bottom left corner, while PERMANOVA R$^2$ and p-values are reported in the top left. Samples from 25 meters deep are visibly distinct from the deep chlorophyll maximum (DCM) samples and an SLA signal is visible in the 25 meter samples only.*

```{r top20compounds stacked barplot, eval=FALSE}
scaled_fk_data %>%
  ungroup() %>%
  arrange(desc(norm_area)) %>%
  filter(feature%in%head(unique(feature), 20)) %>%
  left_join(filled_file_metadata) %>%
  left_join(untarg_feats) %>%
  mutate(feature=ifelse(is.na(compound_name), feature, compound_name)) %>%
  mutate(feature=fct_inorder(feature)) %>%
  arrange(depth, filename) %>%
  mutate(filename=str_remove(filename, "190715_Smp_FK180310")) %>%
  mutate(filename=str_remove(filename, "-25m_|-DCM_")) %>%
  mutate(filename=fct_inorder(filename)) %>%
  ggplot() +
  geom_col(aes(x=filename, y=norm_area, fill=feature)) +
  theme(axis.text.x = element_text(angle=90, hjust = 1, vjust=0.5)) +
  facet_wrap(~depth, scales="free_y", ncol=1)
```

When analyzed separately as distinct depths, the 25 meter samples had higher variances explained by eddy and a lower p-value (PERMANOVA, R$^2$=0.26, p-value=0.013), while the DCM samples were no longer likely to be distinct between the two eddies (PERMANOVA p-value = 0.171) (Figure 6, Supplemental Figure 6).

```{r falkor concentrations comparison to MESO-SCOPE, eval=FALSE}
fk_targ <- targ_peaks %>% 
  left_join(filled_file_metadata) %>%
  filter(ms_run%in%c("FK", "MT")) %>%
  filter(samp_type=="Smp") %>%
  select(compound_name, filename, nM, depth, ms_run)
fk_targ %>%
  group_by(depth, compound_name, ms_run) %>%
  summarise(nM=mean(nM)) %>%
  arrange(desc(nM)) %>%
  pivot_wider(names_from = c(ms_run, depth), values_from = nM) %>%
  print(n=Inf)
cmpd_comp_gp <- fk_targ %>%
  filter(depth!="Deep") %>%
  ggplot() +
  geom_boxplot(aes(x=ms_run, y=nM)) +
  facet_grid(compound_name~depth, scales="free_y")
ggsave("cmpd_comp_gp.pdf", cmpd_comp_gp, device = "pdf", width = 6, height = 50, limitsize = FALSE)
# Visual assessment of boxplots from the above output:
# Compounds more abundant in Falkor: citrulline, aspartic acid, ornithine, threonine, tyrosine, malic acid
# Compounds more abundant in MESO: 4-aminobutyric, adenine, gonyol, 4-hydroxyisoleucine, alanine, leucine, nicotinic acid
```

We were also able to use the HEE dataset to test whether the compounds detected as significantly different in the MESO-SCOPE dataset were also different in this eddy pair. We expected to find abundant TMAO and more hydroxyisoleucine at the surface than the DCM in addition to most compounds enriched in the cyclone DCM with biomass, especially the six compounds mentioned above in the results of Figure 5 (trigonelline, homarine, DMS-Ac, DMSP, taurine, and isethionic acid).

```{r abundant TMAO and hydroxyisoleucine depth trends, eval=FALSE}
fk_targ <- targ_peaks %>% 
  left_join(filled_file_metadata) %>%
  filter(ms_run%in%c("FK")) %>%
  filter(samp_type=="Smp") %>%
  select(compound_name, filename, nM, depth, ms_run)

fk_targ %>%
  filter(compound_name=="Hydroxyisoleucine") %>%
  left_join(filled_file_metadata) %>%
  aov(formula=nM~depth) %>% summary()
  # ggplot(aes(x=depth, y=nM, color=station)) +
  # geom_jitter(width=0.2)
  # geom_boxplot()

fk_targ %>%
  filter(compound_name%in%c("Homarine", "Trigonelline", "Dimethylsulfonioacetate", "Dimethylsulfoniopropionate", "Taurine", "Isethionic acid", "Arsenobetaine")) %>%
  left_join(filled_file_metadata) %>%
  group_by(filename) %>%
  mutate(nM=nM/sum(nM)) %>%
  ungroup() %>%
  filter(depth=="DCM") %>%
  nest(data=-compound_name) %>%
  mutate(aovout=map(data, ~broom::tidy(aov(.x$nM~.x$sla_class)))) %>%
  unnest(aovout) %>% filter(term==".x$sla_class") %>%
  mutate(p.value=p.adjust(p.value, method="fdr"))
  ggplot(aes(x=sla_class, y=nM)) +
  # geom_boxplot() +
  geom_point() +
  facet_wrap(~compound_name, scales="free_y")

untarg_feats %>%
  filter(mzmed%between%pmppm(170.11756, 5) | mzmed%between%pmppm(173.09211, 5)) %>%
  filter(startsWith(feature, "FK")) %>% 
  left_join(untarg_peaks) %>%
  left_join(filled_file_metadata) %>%
  nest(data=-feature) %>%
  mutate(aovout=map(data, ~broom::tidy(aov(.x$norm_area~.x$sla_class)))) %>%
  unnest(aovout) %>% filter(term==".x$sla_class") %>%
  mutate(p.value=p.adjust(p.value, method="fdr"))
  select(feature, sla_class, norm_area) %>% 
  pivot_wider(names_from = sla_class, values_from = norm_area, values_fn = mean)
  ggplot(aes(x=sla_class, y=norm_area)) +
  # geom_boxplot() +
  geom_point() +
  facet_wrap(~feature, scales="free_y")
```

As expected, TMAO was again one of the most abundant metabolites detected in the particulate matter with concentrations around 1.2 nM at 25 meters and 0.4 nM at the DCM, second only to the high levels of the amino acid glycine (2 nM at 25 meters, 0.9 nM DCM), though the differences between 25 meters and DCM were not significant for either compound. We also detected a significant difference between depths for hydroxyisoleucine (25 meter mean = 0.32 nM, DCM mean = 0.21 nM, t-test p-value=0.022 with n=12 samples at each depth). These results imply that the overall depth structure of the metabolome of the gyre is relatively fixed for the most abundant compounds.

The eddy effects, on the other hand, were much less strong during this cruise. Of the six known metabolites with the strong enrichment in the cyclone at the DCM in the MESO-SCOPE cruise, only isethionic acid was significantly different in the HEE dataset (isethionic acid t-test p-value$_{FDR}$ = 0.043, other five p-values > 0.25). This difference in isethionic acid concentration was only found after normalizing each sample to the sum of all metabolites in the sample to control for biomass. Arsenobetaine was again found to be strongly enriched in the anticyclone at the DCM both when normalizing to biomass and when not, just as in MESO-SCOPE. Surprisingly, the 173.0921 *m/z* mass feature was also slightly enriched but this time in the *cyclone* DCM with peak areas approximately 1.4 times the values in the anticyclone (t-test p-value = 0.067, Mann-Whitney p-value = 0.045). The 170.1176 *m/z* mass feature was not detected in the HEE samples.

# Discussion

## Multivariate approaches reveal metabolome-wide shifts across pairs of eddies of opposite polarity

We measured the metabolome of samples collected across two sets of adjacent eddies of opposite polarity to explore the effect of sea level anomaly (SLA) on metabolite composition and found that the altered biogeochemistry and microbial community composition between cyclonic and anticyclonic eddies explain a significant portion of the observed variations in their particulate metabolomes.

In all of the datasets analyzed here, we detected a significant difference in the composition of the metabolome between the adjacent eddies. This effect was strongest in the samples taken during the Lagrangian stations at the center of each eddy in the 2017 MESO-SCOPE cruise, with nearly a quarter of the total variance explained by the eddy from which the samples were taken. In the samples taken along the eddy transect, the largest effect was detected in the deepest samples taken from 175 meters and from the DCM, with much less of a response at 25 meters. This lack of SLA effect at the surface is unsurprising given the similarity in the biogeochemistry of the well-lit upper ocean previously reported [@Barone2022]. In the HEE samples, however, the opposite was true with a surprisingly larger SLA response at 25 meters than that at the DCM. In each case, the differences introduced by SLA were smaller than those of depth when compared directly, with even the high-resolution sampling around the DCM finding slightly more variance explained by the 40 meter difference in sampling depth than the 40 centimeter difference in SLA. This result agrees well with prior research showing the large effect of sampling depth on the metabolome [@Heal2021; @Kumler2023; @Bent2024]. Time of day was included as a significant factor in the MESO-SCOPE transect, with trehalose and sucrose in particular showing large differences between 6 AM and 6 PM as has been previously noted [@Muratore2022], though this effect was diminished at 175 meters.

Although possible that the observed differences in metabolomes across adjacent eddies differing in polarity are due to latitudinal shifts or simply background variation in the gyre environment, the transect data in particular implies that this is unlikely to be the case. Samples taken when the absolute value of the SLA was less than 5 cm were more similar to each other than to those samples taken within the eddies despite larger differences in latitude. The transect sampled both outside of each eddy and between the two across a large spatial gradient (~4° latitude and 2° longitude), yet the non-eddy (absolute value of SLA < 5 cm) samples from all three locations grouped together, showed similar median metabolite concentrations, and were visually similar in the most abundant metabolite composition. These results imply that strong cyclonic and anticyclonic eddies represent endpoints in NPSG composition, in agreement with results from @Barone2019 which found the most extreme values in the Hawaii Ocean Time series data typically detected when eddies passed over Station ALOHA. We also detected the largest differences in the DCM metabolome at the exact center of the eddy, with Station 12 highly distinct from even the nearest stations, while no such stark difference was found in the anticyclone (Figures 1 and 2). This indicates that the exact center of the cyclone has a unique metabolome at the DCM relative to the rest of the transect, rather than existing as a smooth gradient with SLA.

Shifts in total biomass were a large driver of metabolomic differences, with earlier work in the same location by @Barone2022 showing that particulate carbon, chlorophyll, and beam attenuation were all 20-80% higher in the DCM of the cyclone relative to the anticyclone. We see this reflected particularly well in the targeted metabolites measured here, with clear depth trends and a strong correlation between total metabolite concentration and particulate carbon values. However, even after controlling for biomass effects by normalizing each sample to the sum of signal measured within it we still found a significant SLA effect, likely due to shifts in the community composition and in particular the increased eukaryote presence in the cyclone DCM. 

```{r multimodel testing maybe, eval=FALSE}
dist_mat <- untarg_peaks %>%
  left_join(filled_file_metadata %>% distinct(ms_run, station, filename, depth), 
            by="filename") %>% filter(ms_run=="MC") %>% 
  mutate(depth=factor(depth, levels=c("DCMless20m", "DCMless10m", "DCM", 
                                      "DCMplus10m", "DCMplus20m"))) %>%
  arrange(feature, station, depth) %>%
  mutate(filename=fct_inorder(filename)) %>%
  group_by(feature) %>%
  mutate(norm_area=rank(norm_area)) %>%
  select(feature, filename, norm_area) %>%
  pivot_wider(names_from=feature, values_from="norm_area") %>%
  mutate(filename=str_remove_all(filename, "180205_Smp_|.mzML")) %>%
  column_to_rownames("filename") %>%
  dist(method = "manhattan") %>%
  as.matrix()
dist_mat %>%
  as.data.frame() %>%
  rownames_to_column("filename_a") %>%
  pivot_longer(-filename_a, names_to="filename_b", values_to = "dist") %>%
  mutate(filename_a=fct_inorder(filename_a)) %>%
  mutate(filename_b=fct_inorder(filename_b)) %>%
  filter(as.numeric(filename_b)>=as.numeric(filename_a)) %>%
  ggplot() +
  geom_raster(aes(x=filename_a, y=filename_b, fill=dist)) +
  coord_fixed() +
  scale_fill_gradient2(midpoint = 2200) +
  theme_minimal() +
  theme(axis.text.x = element_text(angle=90, hjust=1, vjust=0.5))

expl_df <- filled_file_metadata %>% 
  distinct(ms_run, station, filename, depth, samp_type) %>%
  left_join(untarg_peaks, by="filename") %>%
  filter(ms_run=="MC") %>%
  filter(samp_type=="Smp") %>%
  mutate(model1_groups=as.numeric(factor(depth))) %>%
  mutate(model2_groups=case_when(
    station=="L1" & depth %in% c("DCMless20", "DCMless10") ~ 1,
    station=="L1" & depth=="DCM" | station=="L2" & depth=="DCMless20" ~ 2,
    station=="L1" & depth=="DCMplus10" | station=="L2" & depth=="DCMless10" ~ 3,
    station=="L1" & depth=="DCMplus20" | station=="L2" & depth=="DCM" ~ 4,
    TRUE ~ 5
  )) %>%
  distinct(filename, model1_groups, model2_groups)

vegan::adonis2(
  dist_mat~model1_groups+model2_groups, data = expl_df, by = "margin", 
  permutations = 999, method = "manhattan")
```

The NMDS plots of the high-resolution depth samples were also illustrative of sample dissimilarity with clear SLA and depth trends. The anticyclonic samples grouped tightly and showed little variance with depth or triplicate when compared to the cyclonic samples with the exception of DCM triplicate B, which was highly distinct for unknown reasons. In contrast, the cyclonic samples were much more variable as is perhaps expected when the biomass is in the form of larger phytoplankton that are less homogenous in the environment. One notable aspect of their clustering was the way in which the deepest samples (DCM plus 20m) grouped most closely to the anticyclonic samples, perhaps indicating that below the DCM the metabolome rapidly approaches a uniform deep-water signal. This result was noted previously in the 175 meter samples of the eddy transect in @Kumler2023, where samples from the deep euphotic zone showed greater intra-depth similarity than samples from the DCM or 25 meters.

Given the confluence of both depth and SLA signals and the way that the median metabolite plots of Figure 2 would confound the SLA signal with the depth variance, we used k-means clustering to group similar compounds and provide a reduced dimensionality space for visualization. This revealed four major patterns of metabolite response, with a majority of compounds responding to eddy state (Clusters 1, 2, and 4 in Figure 5). It is interesting to note that the sole cluster in which abundances in the anticyclone are greater than those of the cyclone is also the only cluster to generally increase in abundance with depth (Cluster 4), likely because compounds more concentrated in the anticyclone are the same kind of degradation products and recalcitrant carbon typically found at depth.

The Hawaiian Eddy Experiment (HEE) data upset many of the expectations we developed during the MESO-SCOPE cruise in the previous year. Most surprising was the large difference between the eddies detected at the *surface*, while the DCM samples were functionally indistinguishable. This contrasts directly with the results from MESO-SCOPE and the results in @Gleich2024, who found eddy-driven shifts in protistan community composition to be larger at depth than at the surface. However, @Dugenne2023 found differences in nitrogen fixation rate and nitrogen fixer composition varying widely throughout the upper water column. The large surface differences were especially surprising given the large inter-replicate differences between the HEE surface samples in this study, while the DCM samples tended to be much more consistent.

## Univariate approaches highlight individual metabolites responding to lifted and depressed isopycnals

```{r kheal comparison, include=FALSE}
raw_kheal <- read_csv("https://raw.githubusercontent.com/kheal/Gradients1_SemiTargeted3/master/Tables/Manuscript_tables/SuppTables/Full_EnviroCulture_PeakAreas.csv")
kheal_envirosamp <- raw_kheal %>%
  filter(Column=="HILIC") %>%
  mutate(polarity=ifelse(z>0, "pos", "neg")) %>%
  distinct(MassFeature_Column, compound_name_kath=Identification, Confidence, mz, rt, polarity) %>%
  mutate(rt=rt/60)

untarg_feats %>%
  filter(str_detect(feature, "FK")) %>%
  select(feature, polarity, mzmed, rtmed, compound_name_will=compound_name) %>%
  mutate(mz_lower=mzmed*(1-5/1e6)) %>%
  mutate(mz_upper=mzmed*(1+5/1e6)) %>%
  full_join(kheal_envirosamp, by=join_by(polarity, mz_lower<mz, mz_upper>mz)) %>%
  filter(compound_name_kath=="Histidine") %>%
  distinct(feature, MassFeature_Column, mzmed, rtmed, mz, rt, compound_name_will, compound_name_kath)

```

Several molecule- or pathway-specific narratives emerged from the metabolomic data. The untargeted approach used here allowed us to describe and characterize signals from small molecules whose identity was unknown, a particularly promising approach in open ocean gyres where the largest fraction of unknowns is found [@Heal2021]. We found that our list of authentic standards covered fewer metabolites in anticyclonic eddies at all depths, with the molecular diversity of the cyclone much better characterized. Many of the trends detected persisted even after normalizing within each sample, indicating that the shifts discussed have implications beyond simple scaling with biomass.

Of those molecules whose identity was known, the clearest response to SLA was that of the enigmatic osmolyte homarine at the DCM. This abundant compound increased approximately threefold in concentration from < 50 pM in the anticyclone to around 150 pM in the center of the cyclone in both the eddy transect and eddy center datasets. The pattern was weaker in the noisier HEE data but the largest concentrations were still detected in the cyclone and the median cyclone measurement was greater than the maximum value for the anticyclone. This molecule has a well-established role as an osmolyte in eukaryotic phytoplankton and cyanobacteria [@Gebser2013; @Dawson2020; @Heal2021; @Durham2022] as well as a documented decrease in abundance with depth [@Heal2021], indicating that its response to SLA is potentially due to differences in physical and chemical attributes that in turn control biomass, community composition, and recycling rates. Curiously, the distribution of homarine was also tightly correlated with that of trigonelline. Although they have structural similarity they are not known to have a relationship beyond their shared function as osmolytes.

```{r homarine by SLA plot, eval=FALSE}
targ_hom <- targ_peaks %>%
  filter(compound_name=="Homarine") %>%
  left_join(filled_file_metadata) %>%
  filter(samp_type=="Smp")
mt_hom_gp <- targ_hom %>%
  filter(ms_run=="MT") %>%
  filter(depth=="DCM") %>%
  ggplot(aes(x=sla_corr, y=nM)) +
  geom_hline(yintercept = 0) +
  geom_smooth(method = "lm", formula="y~x", color="black") +
  geom_point(aes(fill=sla_corr), pch=21, size=4) +
  scale_fill_gradient2(
    low = scales::muted("blue"), high = scales::muted("red"),  mid = "grey90",
    aesthetics = "fill", guide = NULL
  ) +
  coord_cartesian(ylim = c(0, 0.24), xlim=c(-15.77, 24.99), expand = FALSE) +
  labs(x="Sea level anomaly (cm)", y="Concentration of homarine (nM)") +
  theme_bw()
mc_hom_gp <- targ_hom %>%
  filter(ms_run=="MC") %>%
  ggplot(aes(x=station, y=nM, color=station)) +
  geom_hline(yintercept = 0) +
  geom_boxplot(linewidth=0.75) +
  coord_cartesian(ylim = c(0, 0.24), xlim=c(0.5, 2.5), expand = FALSE) + 
  scale_x_discrete(breaks = c("L1", "L2"), labels = c("Cyclone\ncenter", "Anticyclone\ncenter")) +
  scale_color_manual(values = c(scales::muted("blue"), scales::muted("red")),
                     breaks = c("L1", "L2")) +
  theme_bw() +
  theme(axis.text.y = element_blank(), axis.title.y = element_blank(),
        legend.position = "none", axis.title.x = element_blank())
egg::ggarrange(mt_hom_gp, mc_hom_gp, nrow=1, widths = c(2, 1))


targ_hom %>%
  filter(ms_run=="FK") %>%
  filter(depth=="Surface") %>%
  # with(t.test(formula=nM~sla_class))
  # with(wilcox.test(formula=nM~sla_class))
  ggplot(aes(x=sla_class, y=nM, color=sla_class)) +
  geom_hline(yintercept = 0) +
  geom_boxplot(linewidth=0.75) +
  coord_cartesian(ylim = c(0, 0.5), xlim=c(0.5, 2.5), expand = FALSE) +
  scale_color_manual(values = c(scales::muted("blue"), scales::muted("red")),
                     breaks = c("Cyclone", "Anticyclone")) +
  theme_bw()

```

Isethionate is a known osmolyte thus far found exclusively in eukaryotes [@Durham2022] and strongly associated with a few diatoms in particular [@Heal2021]. This compound was enriched in the cyclone of the DCM in both the MESO-SCOPE and HEE cruises along with its precursor taurine, validating earlier results from @Barone2022 that showed strong enrichment of eukaryotic phytoplankton at the DCM and *Pseudo-nitzschia* in particular. Additionally, the abundance other sulfur-containing compounds (e.g. dimethylsulfonioacetate and dimethylsulfoniopropionate) in the cyclone may indicate links between the formation of eukaryote-rich cyclonic features and the production of dimethylsulfide, a climate-active volatile gas [@Moran2019].

```{r isethionate and taurine, eval=FALSE}
targ_peaks %>%
  filter(compound_name%in%c("Isethionic acid", "Taurine")) %>%
  filter(filename!="180821_Smp_MS11C215m_A.mzML") %>%
  left_join(filled_file_metadata) %>%
  filter(samp_type=="Smp") %>%
  mutate(depth=fct_inorder(depth)) %>%
  mutate(ms_run=factor(ms_run, levels=c("MT", "MC", "FK"))) %>%
  mutate(sla_class=factor(sla_class, levels=c("Anticyclone", "Neither", "Cyclone"))) %>%
  # group_by(compound_name, ms_run, sla_class, depth) %>%
  # summarise(nM=mean(nM)) %>%
  # pivot_wider(names_from = sla_class, values_from = nM)
  ggplot() +
  geom_boxplot(aes(x=depth, y=nM, color=sla_class)) +
  facet_grid(compound_name~ms_run, scales="free")
```

The strong SLA signal detected among the 175 meter samples was partially surprising to us given our expectation about the strongest effect at the DCM where eddy effects lift large concentrations of nutrients above the 1% light level. Instead, we found more compounds overall to be significantly different across the eddy transect in the 175 meter samples than we did at the DCM (α=0.05, 43 compounds at 175 meters vs 22 at the DCM). This was likely driven by the enhancement of organic matter and heterotrophic picoplankton below the DCM as seen in @Barone2019, with the enrichment of nucleobases in the 175 meter samples additionally hinting at increases in bacterial biomass given that nucleobases tend to be highly abundant in bacterial cultures [@Heal2021]. The abundance of a mass feature putatively identified as acetylcholine in the 175 meter cyclone samples followed the same general trend as arsenobetaine, both enriched at depth and in the anticyclone, and may be another compound that results from heterotrophic degradation of phytoplankton metabolites [@Durham2022]. These results raise important questions about the depth at which the community is isolated from SLA effects, with additional data from @Barone2022 and @Gleich2024 indicating that even at 250 meters eddy effects are still discernable.


## Metabolites link changes in biogeochemistry to community biology

Missing links between the environment with its associated community and the composition of particulate matter make it difficult to estimate important controls on the marine carbon cycle. This is especially true for dynamic and short-lived environments such as eddies. Metabolites offer a way to bridge this gap in knowledge by directly measuring the units of exchange between organisms and their environment.

Here, we saw how changes in the density surface driven by physics during this process led to the coupling of physics, chemistry, and biology. The enrichment of eukaryotic organisms at the DCM of the cyclone mirrors other ocean processes where upwelling of deep, nutrient-rich water leads to increases in nitrogen-rich metabolites representative of a shift towards organisms such as diatoms and dinoflagellates that undergo bloom dynamics. @Barone2022 established a "chemical wake" hypothesis in which cyclone formation introduces a transient phytoplankton community with net photosynthetic production that induces export, a conceptual model that can be extended to include many of the metabolites named here, as some metabolites are more bioavailable than others and the "flavor" of this chemical wake will change over time.

Almost all large phytoplankton classes in the 2017 eddy pair were enriched in the cyclone DCM relative to the anticyclone (particularly prymnesiophytes, dinoflagellates, diatoms, and green algae), while *Prochlorococcus* was significantly less abundant and *Synechococcus* was approximately equal [@Barone2022]. Thus, the metabolite data presented here does not help distinguish between large phytoplankton classes but instead provides additional cues about the shift from a metabolome dominated by cyanobacteria producers into one dominated by eukaryotic organisms. This shift occurs anywhere

A shift from targeted to untargeted metabolites may also in part explain why the increased biomass in the cyclone did not result in large increases in export [@Barone2022; @Bent2024]. Our list of known compounds covers a large fraction of the small molecules common to most organisms, indicating that unknown molecules are more likely to be rare and therefore more difficult to degrade. If the increased biomass in the cyclone is dominantly in the form of highly bioavailable compounds, it is not surprising that much of it can be easily degraded. This hypothesis is supported by @Gleich2024, who found that heterotrophy-associated transcripts were more abundant in the 2018 cruise cyclone than the anticyclone.

Both cruises occurred in weakening cyclones and stable anticyclones, limiting the degree to which the results presented here can be extrapolated to the full life cycle of an eddy [@Dugenne2023]. Differences between the 2017 and 2018 cruises in both biogeochemistry and community biology additionally make it difficult to generalize the links between these two factors via metabolites and establish the demand for additional characterization of the metabolome in the North Pacific Subtropical Gyre.

# Conclusion

This study reports how changes in the biogeochemistry and community composition due to mesoscale eddies affect the metabolome and thereby the composition of particulate organic matter. A transect across adjacent mesoscale eddies of opposite polarity showed that many metabolites track closely with metrics of overall biomass, with eddy effects stronger at 175 meters than at 15 meters or the deep chlorophyll maximum (DCM). High-resolution depth sampling of the DCM at the center of each eddy elucidated several known and unknown biomarkers likely corresponding to the previously documented increase in eukaryotic phytoplankton in the cyclone. Metabolite clusters aligned well with expected trends in abundance with depth and between the two eddies, with convergence below the DCM towards a general deep-sea metabolite signal for many metabolites. By contrasting these effects with a follow-up analysis in the centers of a separate pair of mesoscale eddies of opposite polarity the following year, we learned that the impact of eddies may be influenced by their origin, age, and period of the year. Finally, the metabolites with the strongest responses to eddy effects were unknown, and a relatively smaller proportion of known metabolites in the anticyclone demonstrates the utility of untargeted metabolomics in exploring environmental variation and the large amount of information that is missed by only investigating known compounds. In this work, we have shown the sensitivity of the metabolome to various environmental and community factors and constrained the importance of eddy effects in models of elemental cycling.

# Acknowledgements

The authors would like to thank the other members of the Ingalls Lab who provided assistance in sample processing, integration, and analysis. We are also grateful to the SCOPE science team (Tara Clemente and Tim Burrell) for sample collection during the Hawaiian Eddy Experiment. We would like to acknowledge the captain and crew of the R/V *Kilo Moana* and R/V *Falkor* during the KM1709 and FK180310 cruises for making this science possible. We would also like to acknowledge the analysts involved in the biogeochemical measurements for their willingness to provide data and answer questions (Eric Grabowski, Karin Bjorkman, and Rhea Foreman). This work was supported by grants from the Simons Foundation (SCOPE Award ID 329108 to AI, SF Award ID 385428 to AI, Postdoctoral Fellowship in Marine Microbial Ecology ID 548565 to WQ) 

The authors have no conflicts of interest to declare.

# Data availability

Data and code used for this manuscript are all available online. Biogeochemical data can be sourced from the Simons SCOPE website at https://scope.soest.hawaii.edu/data/ and metabolomics data has been uploaded to Metabolomics Workbench under Project ID PR001738, where the data can be accessed directly via its DOI: 10.21228/M82719. Scripts used to process metabolomics data are available on GitHub at https://github.com/wkumler/MesoscopeMetabolomicsManuscript.

# References

```{r packages, eval=FALSE}
packages_used <- list.files("..", pattern = "*\\.R$|*\\.Rmd$", recursive = TRUE) %>%
  str_subset("old|scripts", negate=TRUE) %>%
  map(function(file_name){
    v <- readLines(paste0("../", file_name))
    libs <- unique(str_extract(v, "(?<=library\\().*?(?=\\)|,)"))
    refs <- unique(str_extract(v, "[[:alnum:]]*(?=::)"))
    pkgs <- unique(c(na.omit(libs), na.omit(refs)))
  }) %>% unlist() %>% unique()
knitr::write_bib(packages_used[nchar(packages_used)>0], file = "r-packages.bib")
```

<div id="refs"></div>

# Supplement
## Supp fig 1: PCA axes 1-3 vs SLA by depth

```{r supp figure/table prcomp vs SLA}
pcdata <- multivar_analysis %>%
  select(depth, pca_output) %>%
  mutate(pca_x=map(pca_output, function(pca_output_i){
    pca_output_i$x[,1:3] %>% as.data.frame()
  })) %>%
  select(depth, pca_x) %>%
  unnest_longer(pca_x, indices_to = "filename") %>%
  unnest_wider(pca_x) %>%
  left_join(filled_file_metadata %>% distinct(filename, sla_corr)) %>%
  pivot_longer(starts_with("PC"))
pcvars <- multivar_analysis %>%
  select(depth, pca_output) %>%
  mutate(perc_expl=map(pca_output, function(pca_output_i){
    var_expl <- pca_output_i$sdev^2
    data.frame(perc_expl=var_expl[1:3]/sum(var_expl), name=paste0("PC", 1:3))
  })) %>%
  unnest(perc_expl) %>%
  mutate(perc_label=paste0(round(perc_expl, digits = 3)*100, "%"))

# Correlation between PCs and SLA at each depth
# pcdata %>%
#   nest(data=-c(depth, name)) %>%
#   mutate(cor_output=map_dbl(data, ~abs(cor(.x$value,.x$sla_corr))))

# Linear model significance testing and parameter report
# pcdata %>%
#   nest(data=-c(depth, name)) %>%
#   mutate(lm_output=map(data, ~broom::tidy(lm(.x$value~.x$sla_corr)))) %>%
#   unnest(lm_output) %>%
#   filter(term==".x$sla_corr") %>%
#   arrange(name, depth) %>%
#   select(depth, name, p.value) %>%
#   split(.$name)

lm_gp <- ggplot(pcdata, aes(x=sla_corr, y=value)) +
  geom_smooth(method="lm", formula=y~x, color="black") +
  geom_point(aes(fill=sla_corr), pch=21, color="black") +
  geom_label(data = pcvars, aes(label=perc_label),
             hjust=1, vjust=1, x=Inf, y=Inf, label.r = unit(0, "in")) +
  facet_grid(depth~name, switch = "y") +
  scale_y_continuous(position = "right") +
  scale_fill_gradient2(
    low = scales::muted("blue"), high = scales::muted("red"),  mid = "grey90",
    aesthetics = c("color", "fill"), guide = NULL
  ) +
  labs(x="Corrected SLA, in cm", y="Principal component value",
       color="Corrected\nSLA (cm)") +
  theme_bw()
deep_pc1_barplot_gp <- pcdata %>%
  left_join(filled_file_metadata %>% distinct(filename, station, tripl)) %>%
  mutate(station=factor(station, levels=as.character(4:14),
                        labels=paste0("St.", 4:14))) %>%
  arrange(station) %>%
  filter(name=="PC1" & str_detect(depth, "Deep")) %>%
  ggplot() +
  geom_hline(yintercept = 0) +
  geom_col(aes(x=tripl, y=value)) +
  facet_wrap(~station, nrow = 1, scales="free_x") +
  scale_y_continuous(position = "right") +
  labs(x="Triplicate", y="PC1", title = paste(
    "    Variance among deep (175 meter) samples in PC1"
  ))
pca_sla_cor_gp <- cowplot::plot_grid(lm_gp, deep_pc1_barplot_gp, 
                                     rel_heights = c(0.6, 0.4), 
                                     ncol = 1, labels = "AUTO")
ggsave("pca_sla_cor_gp.png", pca_sla_cor_gp, device = "png", path = "supplement",
       width = 6.875, height = 6.5, units = "in", dpi = 300)

```

![](supplement/pca_sla_cor_gp.png)

*Supplemental Figure 1: Results of a principal component analysis performed separately on each depth of the eddy transect metabolome. A) Regressions of SLA against various principal components, broken down by depth. % variance explained by each PC is noted in the upper right corner of the plot. Colors have been added corresponding to the SLA value for clarity but are redundant with the x-axis values. Lines of best fit and the associated standard error have been added behind the points in black and translucent grey, respectively. Strong associations exist between PC1 and SLA in the DCM samples and PC2 and SLA in the 175 meter samples. B) Principal component 1 values for the 175 meter samples at each station showing large variance between triplicates rather than any external factor.*

## Supp fig 2: Biomass regression

```{r Fig 2.5 PC vs tot nM plot}
pc_totalnm_plot <- no_outlier_frac_pc_in_metab_df %>%
  ggplot(aes(x=PC_um, y=total_nM_C)) +
  geom_smooth(method="lm", color="white", lwd=2.5, se=FALSE) +
  geom_smooth(method="lm", color="black", lwd=1.5) +
  geom_point(aes(fill=depth), pch=21, size=3) +
  geom_smooth(aes(group=depth), method="lm", color="white", lwd=2.5, se = FALSE) +
  geom_smooth(aes(color=depth), method="lm", lwd=1.5) +
  geom_vline(xintercept = 0) +
  geom_hline(yintercept = 0) +
  theme_bw() +
  theme(legend.position = c(0,1), legend.justification = c(0, 1), legend.background = element_rect(color="black")) +
  labs(color="Sampling depth",
       fill="Sampling depth",
       x="Particulate carbon (μM)", 
       y="Total nM carbon in known metabolites") +
  scale_color_manual(breaks=c("Surface", "DCM", "Deep"), 
                     labels=c("15 m", "DCM", "175 m"),
                     values = c("#F4BB23", "#028E34", "#0B505C"), aesthetics = c("fill", "color"))

pn_totalnm_plot <- no_outlier_frac_pc_in_metab_df %>%
  ggplot(aes(x=PN_um, y=total_nM_N)) +
  geom_point(aes(fill=depth), pch=21, size=3, alpha=0.8) +
  geom_smooth(method="lm", color="white", lwd=2.5, se=FALSE) +
  geom_smooth(method="lm", color="black", lwd=1.5) +
  geom_smooth(aes(group=depth), method="lm", color="white", lwd=2.5, se = FALSE) +
  geom_smooth(aes(color=depth), method="lm", lwd=1.5) +
  geom_vline(xintercept = 0) +
  geom_hline(yintercept = 0) +
  theme_bw() +
  theme(legend.position = "none") +
  labs(color="Sampling depth",
       fill="Sampling depth",
       x="Particulate nitrogen (μM)", 
       y="Total nM nitrogen in known metabolites") +
  scale_color_manual(breaks=c("Surface", "DCM", "Deep"), 
                     labels=c("15 m", "DCM", "175 m"),
                     values = c("#F4BB23", "#028E34", "#0B505C"), aesthetics = c("fill", "color"))


library(ggh4x)
strip_theme <- strip_themed(background_x = elem_list_rect(fill= c("#F4BB23", "#028E34", "#0B505C")))

pc_totalnm_plot_bydepth <- no_outlier_frac_pc_in_metab_df %>%
  mutate(depth=factor(depth, levels=c("Surface", "DCM", "Deep"),
                      labels=c("15 m", "DCM", "175 m"))) %>%
  ggplot() +
  aes(x=PC_um, y=total_nM_C) +
  geom_smooth(method="lm", color="black") +
  geom_point(aes(fill=sla_corr), pch=21, color="black", size=3, alpha=0.8) +
  ggh4x::facet_wrap2(~depth, nrow=1, scales="free", strip = strip_theme) +
  labs(x="Particulate carbon (μM)", y="Total nM carbon in\nknown metabolites") +
  scale_fill_gradient2(
    low = scales::muted("blue"), high = scales::muted("red"),  mid = "grey90",
    aesthetics = c("fill"), name="Corr.\nSLA\n(cm)"
  ) +
  theme_bw() +
  theme(strip.text = element_text(face = "bold", color="white")) +
  guides(fill=guide_colorbar(title.hjust = 0.5, barheight = 5, title.position = "bottom"))


comb_pcpn_plot <- egg::ggarrange(pc_totalnm_plot, pn_totalnm_plot, nrow=1, draw = FALSE)

comb_pcpnnm_plot <- cowplot::plot_grid(comb_pcpn_plot, pc_totalnm_plot_bydepth, ncol = 1,
                                     rel_heights = c(5, 3))

ggsave("comb_pcpnnm_plot.png", plot = comb_pcpnnm_plot, device = "png", 
       path = "supplement", width = 6.875, height = 5, units = "in", dpi = 300, 
       bg = "white")
```

![](supplement/comb_pcpnnm_plot.png)

*Supplemental Figure 2: Type I linear regressions of particulate carbon and nitrogen against total nM carbon and nitrogen in known metabolites both as a whole and divided by depth. Points have been colored according to depth (top row of plots) or SLA (bottom row of plots) and have been placed on independent axes. Two outlier particulate carbon points have been interpolated from beam attenuation for 15 meter values at stations 10 and 11.*

## Supp fig 3: Stacked barplot of top metabs by triplicate

```{r targ barplot relative and absolute}
targ_barplot_df <- targ_peaks %>%
  left_join(filled_file_metadata, by="filename") %>%
  filter(ms_run=="MT") %>%
  filter(samp_type=="Smp") %>%
  group_by(compound_name) %>%
  mutate(mean_cmpd_nm=mean(nM)) %>%
  ungroup() %>%
  arrange(desc(mean_cmpd_nm)) %>%
  mutate(compound_name=ifelse(compound_name=="Trimethylamine N-oxide", "TMAO", compound_name)) %>%
  mutate(compound_name=ifelse(compound_name=="Hydroxyisoleucine", "HO-Ile", compound_name)) %>%
  mutate(compound_name=ifelse(compound_name=="Glycine betaine", "GBT", compound_name)) %>%
  mutate(compound_name=str_remove(compound_name, "^L-")) %>%
  mutate(compound_name=ifelse(
    compound_name%in%head(unique(compound_name), 9), 
    compound_name, "Other")) %>%
  mutate(compound_name=fct_inorder(compound_name)) %>%
  group_by(depth, station, compound_name, tripl) %>%
  summarise(nM=sum(nM), .groups = "keep") %>%
  mutate(depth=factor(depth, levels=c("Surface", "DCM", "Deep"), 
                      labels=c("15 m", "DCM", "175 m"))) %>%
  mutate(station=factor(station, levels=as.character(4:14)))
base_tripl_barplot <- targ_barplot_df %>%
  ggplot() +
  aes(x=tripl, y=nM, fill=compound_name) +
  geom_hline(yintercept = 0) +
  facet_grid(depth~station, scales="free_y", switch = "x") +
  scale_fill_manual(values = c(splividis(9), "grey50")) +
  labs(x="Station", y="Metabolite concentration (nM)", fill=NULL) +
  theme_bw() +
  theme(axis.text.x = element_blank(), axis.ticks.x = element_blank(),
        strip.background.x = element_blank(), 
        panel.border = element_blank())
abs_targ_bar_gp <- base_tripl_barplot +
  geom_col(width=1) +
  ggtitle("A) Absolute concentrations of most abundant metabolites") +
  scale_y_continuous(expand = expansion()) +
  theme(legend.position="none", strip.text.x = element_blank(), axis.title.x = element_blank())
rel_targ_bar_gp <- base_tripl_barplot +
  geom_col(width=1, position = "fill") +
  scale_y_continuous(labels = scales::label_percent(), expand = expansion(0.1)) +
  theme(legend.position = "bottom") +
  guides(fill=guide_legend(nrow=2, byrow = TRUE)) +
  labs(y=NULL) +
  ggtitle("B) Relative composition of the metabolome")
comb_targ_rel_gp <- egg::ggarrange(abs_targ_bar_gp, rel_targ_bar_gp, ncol=1, draw=FALSE)

ggsave("supplement/barplot_comb_targ_rel_gp.png", comb_targ_rel_gp, device = "png", width = 6.875, height = 6.5, units = "in", dpi = 300, bg="white")
```

![](supplement/barplot_comb_targ_rel_gp.png)

*Supplemental Figure 4: Stacked barplots of known metabolite concentration as shown in main text Figure 3 but with individual triplicates shown. The same data is shown in A) as B) but B) is rendered in relative contribution space instead of absolute concentration. TMAO = trimethylamine N-oxide, HO-Ile = hydroxyisoleucine, GBT = glycine betaine, DCM = deep chlorophyll maximum.*

## Supp fig 4: Choline, phosphocholine, and glycerophosphocholine ratios

```{r}
gp_df <- data.frame(
  compound_name=c("Ethanolamine", "Glycerol", "Choline"),
  mz=c(62.060589, 93.05462, 104.107539)
) %>%
  expand_grid(data.frame(
    addition=c("", "Phospho", "Glycerophospho"),
    mz_diff=c(0, 79.96633, 74.03678+79.96633)
  )) %>%
  mutate(base_compound=compound_name) %>%
  mutate(compound_name=ifelse(addition=="", compound_name, 
                              paste0(addition, tolower(compound_name))),
         mz=mz+mz_diff) %>%
  arrange(mz) %>%
  mutate(compound_name=fct_inorder(compound_name)) %>%
  mutate(addition=fct_inorder(addition))
gp_w_eics <- gp_df %>%
  mutate(eic=map(mz, function(mz_i){
    open_dataset("~/My Drive/AllMeso/tmzMLs/MT_all_file_arr_pqds") %>%
      filter(mz%between%pmppm(mz_i, 10)) %>%
      collect()
  }))
smp_eics <- gp_w_eics %>% 
  select(-mz) %>% 
  unnest(eic) %>% 
  filter(str_detect(filename, "Smp")) %>%
  left_join(filled_file_metadata)
all_cmpds <- tribble(
  ~compound_name, ~rt_start, ~rt_end,
  "Choline", 10.6, 11.2,
  "Phosphocholine", 11.8, 13.2,
  "Glycerophosphocholine", 10.8, 11.5,
  "Glycerophosphoethanolamine", 11, 12.5,
  "Glycerophosphoglycerol", 10.2, 11.2
) %>%
  mutate(compound_name=fct_inorder(compound_name)) %>%
  left_join(smp_eics, by=join_by(compound_name, between(y$rt, x$rt_start, x$rt_end))) %>%
  left_join(filled_file_metadata) %>%
  mutate(depth=factor(depth, levels=c("Surface", "DCM", "Deep"), labels=c("15m", "DCM", "175m")))
```

```{r}
gpc_chrom_gp <- all_cmpds %>%
  mutate(compound_name=str_replace(compound_name, "Glycerophospho", "Glycerophospho-\n")) %>%
  mutate(compound_name=fct_inorder(compound_name)) %>%
  mutate(int=int/1e6) %>%
  ggplot() +
  geom_line(aes(x=rt, y=int, group=filename, color=depth)) +
  facet_wrap(~compound_name, scales="free", nrow=1) +
  scale_color_manual(values=viridis::viridis(9, direction = -1)[c(2,5,8)], 
                     breaks = c("15m", "DCM", "175m"),
                     aesthetics = c("color", "fill"),
                     name="Sample\ndepth") +
  theme_bw() +
  labs(x="Retention time (minutes)", y="Intensity (millions)") +
  guides(color=guide_legend(override.aes = list(lwd=2)))
gpc_area_gp <- all_cmpds %>%
  summarise(area=trapz(rt, int)/1e6, .by = c(compound_name, filename, depth, station, sla_corr)) %>%
  mutate(compound_name=str_replace(compound_name, "Glycerophospho", "Glycerophospho-\n")) %>%
  # mutate(station=factor(station, levels=as.character(4:14))) %>%
  mutate(station=as.numeric(station)) %>%
  mutate(compound_name=fct_inorder(compound_name)) %>%
  ggplot() +
  geom_hline(yintercept = 0) +
  geom_point(aes(x=station, y=area, fill=sla_corr), color="black", pch=21) +
  facet_grid2(depth~compound_name, scales="free_y", independent = "y") +
  labs(x="Station number", y="Peak area (millions)", fill="Corr. SLA\n(cm)") +
  scale_color_gradient2(low = scales::muted("blue"), 
                        high = scales::muted("red"),
                        mid="grey90", aesthetics = c("color", "fill")) +
  theme_bw()


gpc_ratio_reg_gp <- all_cmpds %>%
  summarise(area=trapz(rt, int)/1e6, .by = c(compound_name, filename, depth, station, sla_corr)) %>%
  pivot_wider(names_from = compound_name, values_from=area) %>%
  pivot_longer(Phosphocholine:last_col()) %>%
  filter(name%in%c("Phosphocholine", "Glycerophosphocholine")) %>%
  mutate(name=paste0("Choline:", name)) %>%
  mutate(name=fct_inorder(name)) %>%
  ggplot(aes(x=sla_corr, y=Choline/value)) +
  geom_smooth(method="lm", formula=y~x, color="black") +
  geom_point(aes(fill=sla_corr), color="black", pch=21) +
  facet_nested(.~name+depth, scales="free_y", independent = "y") +
  labs(x="Corrected SLA (cm)", y="Metabolite ratio", fill="Corr. SLA\n(cm)") +
  scale_fill_gradient2(low = scales::muted("blue"), 
                        high = scales::muted("red"),
                        mid="grey90", guide = "none") +
  theme_bw()

comb_gpc_patch <- gpc_chrom_gp / gpc_area_gp / gpc_ratio_reg_gp + plot_annotation(tag_levels = 'A') + plot_layout(heights = c(0.3, 0.4, 0.3))
ggsave("supplement/comb_gpc_patch.png", comb_gpc_patch, device = "png", width = 9, height = 6.5, units = "in", dpi = 300, bg="white")
```

![](supplement/comb_gpc_patch.png)
*Supplemental Figure 4: Evidence against phosphorus limitation across the MESO-SCOPE eddy transect. Top row of plots (A) correspond to extracted ion chromatograms for 5 metabolites useful for estimating the phosphorus stress of an environment, colored by the depth from which the sample was taken. Middle row of plots shows the integrated peak area for each metabolite across the 11 stations of the transect with sub-panels for each depth independently, with each point colored by the corrected sea level anomaly (SLA). Bottom row of plots shows the ratio of choline to phosphocholine (left) and choline to glycerophosphocholine (right) as a function of corrected SLA with no statistically significant trend visible at any of the three depths.*

## Supp fig 5: Boxplots of selected compounds by depth, time, and SLA

```{r diel, include=FALSE}
filled_file_metadata %>% 
  filter(samp_type=="Smp") %>%
  filter(ms_run=="MT") %>%
  distinct(station, time_class) %>%
  count(time_class)
filled_file_metadata %>% 
  filter(samp_type=="Smp") %>%
  filter(ms_run=="MT") %>%
  distinct(station, sla_class) %>%
  count(sla_class)

select_sla_time_boxplot_gp <- targ_peaks %>%
  filter(compound_name%in%c("Sucrose", "Trehalose", "Guanine", "Glycine betaine", "L-Glutamine", "Ectoine")) %>%
  mutate(compound_name=case_when(
    compound_name=="Glycine betaine"~"Glycine\nbetaine",
    compound_name=="L-Glutamine"~"Glutamine",
    TRUE~compound_name
  )) %>%
  select(filename, compound_name, nM) %>%
  left_join(filled_file_metadata) %>%
  filter(samp_type=="Smp") %>%
  filter(ms_run!="MC") %>%
  select(filename, compound_name, cruise, depth, sla_class, time_class, nM) %>%
  mutate(time_class=factor(time_class, levels=c("Morning", "Midday", "Evening", "Night"))) %>%
  mutate(cruise=factor(cruise, levels=c("MS", "FK"), labels=c("MESO-SCOPE transect", "HEE"))) %>%
  mutate(depth=factor(depth, levels=c("Surface", "DCM", "Deep"))) %>%
  ggplot() +
  geom_hline(yintercept = 0) +
  geom_boxplot(aes(x=time_class, y=nM, color=sla_class)) +
  facet_nested(compound_name~cruise+depth, scales="free", space = "free_x") +
  labs(x=NULL, y="Metabolite concentration (nM)", color="SLA condition") +
  scale_color_manual(breaks = c("Cyclone", "Neither", "Anticyclone"),
                     values = c(scales::muted("blue"), "grey50", 
                                scales::muted("red"))) +
  theme_bw() +
  theme(axis.text.x = element_text(angle=90, hjust=1, vjust=0.5))

ggsave("diel_effect_boxplots.png", plot = select_sla_time_boxplot_gp, device = "png",
       path = "supplement", width = 6.875, height = 6, units = "in", dpi = 300,
       bg="white")
```

![](supplement/diel_effect_boxplots.png)

*Supplemental Figure 5: Boxplots showing the concentrations of six selected compounds between cruises and depths as well as time and SLA classes. The abscissa is separated by the time category into which sampling was binned, referring to 6-hour increments starting from 3AM (e.g. Midday contains casts collected between 9AM and 3PM). Colors denote the sea level anomaly bins with a ±5cm cutoff. For the eddy transect, six stations are included in the anticyclone, three in the cyclone, and two were neither. There were two morning stations, 4 midday stations, 2 evening stations, and 3 night stations. Diel effects are clearly visible for sucrose and trehalose while other compounds show very little diel differences. Trehalose was not quantified during the MESO-SCOPE transect for lack of an authentic standard to quantify the response factor.*

## Supp fig 6: HEE NMDS and PERMANOVA broken down by depth


```{r falkor nmdsanova by depth}
set.seed(123)
multivar_analysis_FK <- untarg_peaks %>%
  filter(filename!="190715_Smp_FK180310S62C1-DCM_B.mzML") %>%
  left_join(filled_file_metadata %>% distinct(ms_run, station, filename, depth), 
          by="filename") %>%
  filter(ms_run=="FK") %>%
  mutate(depth=factor(depth, levels=c("Surface", "DCM"))) %>%
  nest(data=-depth) %>%
  mutate(wide_peaks=map(data, function(metab_df){
    metab_df %>%
      select(feature, filename, norm_area) %>%
      group_by(feature) %>%
      filter(sum(norm_area==0)<n()*0.9) %>%
      mutate(norm_area=scale(norm_area)[,1]) %>%
      pivot_wider(names_from = "feature", values_from = norm_area) %>%
      column_to_rownames("filename") %>%
      data.matrix()
  })) %>%
  mutate(nmds_output=map(wide_peaks, function(metab_mat){
    env_vec <- pull(filled_file_metadata, sla_corr, filename)[rownames(metab_mat)]
    vegan::metaMDS(metab_mat, k = 2, distance = "manhattan", trace = 0, 
                   autotransform = FALSE, noshare = FALSE, wascores = FALSE) %>%
    vegan::MDSrotate(env_vec)
  })) %>%
  mutate(nmds_stress=map_dbl(nmds_output, pluck, "stress")) %>%
  mutate(perm_output=map(wide_peaks, function(metab_mat){
    adon_expl <- data.frame(filename=rownames(metab_mat)) %>%
      left_join(filled_file_metadata %>% select(filename, sla_class), by="filename")
    vegan::adonis2(
      metab_mat~sla_class, data = adon_expl, by = "margin", 
      permutations = 999, method = "manhattan")
  })) %>%
  mutate(perm_r2=map_dbl(perm_output, pluck, "R2", 1)) %>%
  mutate(perm_pval=map_dbl(perm_output, pluck, "Pr(>F)", 1)) %>%
  mutate(pca_output=map(wide_peaks, prcomp)) %>%
  mutate(nmds_stress=paste("s =", round(nmds_stress, digits = 3))) %>%
  mutate(perm_r2=round(perm_r2, digits = 2)) %>%
  mutate(perm_pval=round(perm_pval, digits = 4)) %>%
  mutate(perm_label=paste0("R2 = ", perm_r2, "\np = ", perm_pval))
nmds_sla_gp <- multivar_analysis_FK %>%
  mutate(nmds_points=map(nmds_output, function(nmds_output_i){
    nmds_output_i %>% pluck("points") %>% as.data.frame()
  })) %>%
  select(depth, nmds_points) %>%
  unnest_longer(nmds_points, indices_to = "filename") %>%
  unnest_wider(nmds_points) %>%
  left_join(filled_file_metadata %>% distinct(filename, sla_corr)) %>%
  ggplot() +
  geom_vline(xintercept = 0) +
  geom_hline(yintercept = 0) +
  geom_point(aes(x=MDS1, y=MDS2, fill=sla_corr), color="black", size=4, pch=21) +
  facet_wrap(~depth, nrow=1) +
  scale_fill_gradient2(
    low = scales::muted("blue"), high = scales::muted("red"),  mid = "grey90",
    aesthetics = c("color", "fill"),
  ) +
  labs(fill="Sea level\nanomaly\n(cm)\n") +
  theme_bw() +
  coord_fixed()

med_metab_df <- untarg_peaks %>%
  left_join(filled_file_metadata %>% distinct(filename, ms_run, station, depth, sla_corr),
            by="filename") %>%
  filter(ms_run=="FK") %>%
  group_by(depth, feature) %>%
  filter(sum(norm_area==0)<n()*0.9) %>%
  mutate(norm_area=scale(norm_area)[,1]) %>%
  ungroup() %>%
  group_by(filename, depth, station, sla_corr) %>%
  summarize(avg_amt=mean(norm_area)) %>%
  ungroup()
# Calculations for correlation between grand median metabolite and SLA
# med_metab_df %>%
#   group_by(station, depth, sla_corr) %>%
#   summarise(grand_avg_amt=mean(avg_amt)) %>%
#   group_by(depth) %>%
#   summarise(sim=cor(sla_corr, grand_avg_amt))
med_metab_sla_gp <- med_metab_df %>%
  mutate(station=factor(station, levels=c("62", "64", "77", "80"))) %>%
  mutate(depth=factor(depth, levels=c("Surface", "DCM"))) %>%
  ggplot() +
  geom_hline(yintercept = 0) +
  geom_boxplot(aes(x=station, y=avg_amt, color=sla_corr), linewidth=0.8) +
  facet_wrap(~depth, nrow=1) +
  scale_color_gradient2(low = scales::muted("blue"), high = scales::muted("red"),
                        mid="grey90", aesthetics = c("color", "fill"), guide = NULL) +
  theme_bw() +
  theme(strip.background = element_blank(), strip.text = element_blank()) +
  labs(x="Station", y="Median metabolite z-score")

library(egg)
nolegend_plot <- ggarrange(nmds_sla_gp+theme(legend.position = "none"), 
                           med_metab_sla_gp, ncol = 1, heights = c(0.7, 0.3), 
                           draw = FALSE)
library(cowplot)
fk_nmds_and_med_metab <- plot_grid(nolegend_plot, get_legend(nmds_sla_gp), 
                                nrow=1, rel_widths = c(0.85, 0.15))

ggsave("fk_nmds_and_med_metab.png", plot = fk_nmds_and_med_metab, device = "png",
       path = "supplement", width = 6.875, height = 5.5, units = "in", dpi = 300,
       bg="white")
```

![](supplement/fk_nmds_and_med_metab.png)

*Supplemental Figure 6: Plots of the metabolome during the Hawaiian Eddy Experiment, broken down by depth. The top row of plots are non-metric multidimensional scaling (NMDS) plots, in which points correspond to individual samples and have been colored by their corrected sea level height anomaly. SLA trends are visible in the 25 meter samples, with dark blue circles consistently discriminating from the dark red circles. The bottom row of plots show the direction and magnitude of this effect by plotting the grand mean of the normalized metabolite peak areas in the stations taken between the adjacent eddies of opposite polarity.*

